<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Workshop: MoTrPAC R Packages from the Endurance Training in Young Rats Study • MotrpacWorkshops</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Workshop: MoTrPAC R Packages from the Endurance Training in Young Rats Study">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MotrpacWorkshops</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/rat-endurance-6m.html">Workshop: MoTrPAC R Packages from the Endurance Training in Young Rats Study</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Workshop: MoTrPAC R Packages from the Endurance Training in Young Rats Study</h1>
            
      

      <div class="d-none name"><code>rat-endurance-6m.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The Molecular Transducers of Physical Activity Consortium (MoTrPAC)
conducted the first comprehensive, organism-wide, multi-omic
investigation of endurance exercise training effects in rats. This
large-scale study involved six-month-old male and female Fischer 344
rats subjected to progressive treadmill endurance exercise training over
1, 2, 4, or 8 weeks. Tissues were collected from these rats 48 hours
after the last exercise bout to capture the adaptations resulting from
the training, with sex-matched sedentary, untrained rats serving as
controls. Utilizing multiple ‘omic’ platforms including transcriptomics,
epigenomics, proteomics, and metabolomics—across a wide range of
tissues, the study provides a rich dataset that captures the temporal
effects of endurance exercise training at multiple biological levels.
This dataset offers invaluable insights into the molecular mechanisms
underlying exercise-induced adaptations. Find the open access <a href="https://www.nature.com/nature/volumes/629/issues/8010" class="external-link">manuscript
published on Nature</a>.</p>
<p>To facilitate access to and analysis of this extensive dataset, two
essential R packages have been developed:</p>
<ul>
<li>
<a href="https://motrpac.github.io/MotrpacRatTraining6moData/" class="external-link">MotrpacRatTraining6moData</a>:
Provides access to the processed data and downstream analysis results
from the MoTrPAC endurance exercise study.</li>
<li>
<a href="https://motrpac.github.io/MotrpacRatTraining6mo/" class="external-link">MotrpacRatTraining6mo</a>:
Offers functions to help retrieve and explore the data, enabling users
to perform analyses and reproduce key findings from the study.</li>
</ul>
<p>In this tutorial, we will explore how to utilize the
<code>MotrpacRatTraining6moData</code> package to access and navigate
the extensive dataset generated by the endurance exercise study, and
employ the <code>MotrpacRatTraining6mo</code> package to perform
analyses and gain insights into the molecular responses to endurance
training in young rats.</p>
<p><strong>Note</strong>: This notebook is based on the vignettes
available in <a href="https://motrpac.github.io/MotrpacRatTraining6moData/articles/MotrpacRatTraining6moData.html" class="external-link">MotrpacRatTraining6moData</a>
and <a href="https://motrpac.github.io/MotrpacRatTraining6mo/articles/MotrpacRatTraining6mo.html" class="external-link">MotrpacRatTraining6mo
vignette</a>.</p>
<div class="section level3">
<h3 id="installations">Installations<a class="anchor" aria-label="anchor" href="#installations"></a>
</h3>
<p>Install both packages with <code>devtools</code>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>timeout<span class="op">=</span><span class="fl">1e5</span><span class="op">)</span> <span class="co"># extend the timeout for downloading large files</span></span>
<span></span>
<span><span class="co"># First the data package, which might take a long time</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"MoTrPAC/MotrpacRatTraining6moData"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Then the analysis package:</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"MoTrPAC/MotrpacRatTraining6mo"</span><span class="op">)</span></span></code></pre></div>
<p>Once we install this package, we can load these and other libraries
used in this tutorial:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://motrpac.github.io/MotrpacRatTraining6moData/" class="external-link">MotrpacRatTraining6moData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MoTrPAC/MotrpacRatTraining6mo" class="external-link">MotrpacRatTraining6mo</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span> <span class="co"># for plots in this tutorial </span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">impute</span><span class="op">)</span>  <span class="co"># For KNN imputation</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Attaching core tidyverse packages</span> ──────────────────────── tidyverse 2.0.0 ──</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">forcats  </span> 1.0.0     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">stringr  </span> 1.5.1</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">lubridate</span> 1.9.4     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">tibble   </span> 3.2.1</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">purrr    </span> 1.0.4     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">tidyr    </span> 1.3.1</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">readr    </span> 2.1.5</span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Conflicts</span> ────────────────────────────────────────── tidyverse_conflicts() ──</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">filter()</span> masks <span style="color: #0000BB;">stats</span>::filter()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">lag()</span>    masks <span style="color: #0000BB;">stats</span>::lag()</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use the conflicted package (<span style="color: #0000BB; font-style: italic;">&lt;http://conflicted.r-lib.org/&gt;</span>) to force all conflicts to become errors</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://lubridate.tidyverse.org" class="external-link">lubridate</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://sjmgarnier.github.io/viridis/" class="external-link">viridis</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: viridisLite</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="study-design-the-pheno-data-object">Study design: the PHENO data object<a class="anchor" aria-label="anchor" href="#study-design-the-pheno-data-object"></a>
</h2>
<p>Details of the experimental design can be found in the <a href="https://static-content.springer.com/esm/art%3A10.1038%2Fs41586-023-06877-w/MediaObjects/41586_2023_6877_MOESM1_ESM.pdf" class="external-link">supplementary
methods of the Nature paper</a>.</p>
<p>Briefly, 6-month-old young adult rats were subjected to progressive
endurance exercise training for 1, 2, 4, or 8 weeks, with tissues
collected 48 hours after the last training bout. Sex-matched sedentary,
untrained rats were used as controls. Whole blood, plasma, and 18 solid
tissues were analyzed using genomics, proteomics, metabolomics, and
protein immunoassay technologies, with most assays performed in a subset
of these tissues. Depending on the assay, between 3 and 6 replicates per
sex per time point were analyzed.</p>
<p>The <strong>PHENO</strong> object represents phenotypic data from the
MoTrPAC endurance exercise training study in six-month-old rats.</p>
<blockquote>
<p><strong>Important</strong>: The endurance study was conducted at two
Preclinical Animal Study Sites (PASS): the University of Iowa and Joslin
Diabetes Center. <em>Phenotypic data include animals from both
sites</em>; however, omics data were generated only from animals
processed at the University of Iowa site to ensure standardization for
molecular assays. Samples from animals at the Joslin site were not used
for omics analyses but have been preserved and are stored in freezers.
It is anticipated that these samples may become available for follow-up
endurance exercise studies through future funding opportunities,</p>
</blockquote>
<p>The PHENO data object is a comprehensive dataset containing
<strong>5,955 rows and 510 variables</strong>, with each row
corresponding to a unique sample identified by <code>viallabel</code>.
The dataset captures detailed information on the animals, their training
regimens, specimen collection, and various physiological metrics. Here
is a summary of the major components:</p>
<div class="section level4">
<h4 id="categories-of-variables-in-the-pheno-dataset">Categories of Variables in the PHENO Dataset<a class="anchor" aria-label="anchor" href="#categories-of-variables-in-the-pheno-dataset"></a>
</h4>
<ol style="list-style-type: decimal">
<li><p><strong>Identifiers</strong>: Unique identifiers such as
<code>pid</code>, <code>bid</code>, <code>viallabel</code>, and
<code>labelid</code> are used to link samples to individual animals and
their specimen labels.</p></li>
<li><p><strong>Animal Information</strong>: Variables such as
<code>registration.d_birth</code>, <code>registration.sex</code>, and
<code>registration.weight</code> provide basic details about the rats,
including birth date, sex, and weight.</p></li>
<li><p><strong>Registration and Housing</strong>: Data about the
arrival, housing conditions, registration
(<code>registration.d_arrive</code>,
<code>registration.cagenumber</code>), and light conditions
(<code>registration.d_reverselight</code>) are included to document the
environmental conditions in which animals were kept.</p></li>
<li><p><strong>Intervention and Randomization</strong>: Information
about intervention groups (e.g., control or training) is captured in
variables like <code>key.intervention</code>,
<code>key.anirandgroup</code>, and <code>key.protocol</code>, indicating
how each rat was assigned to different treatment conditions.</p></li>
<li><p><strong>Familiarization and Training Data</strong>: Detailed
records of treadmill familiarization
(<code>familiarization.d_treadmillbegin</code>) and progressive
endurance training are included. Training data cover up to <strong>40
days</strong>, with variables for each day’s treadmill speed, incline,
and exercise time (<code>training.dayX_treadmillspeed</code>,
<code>training.dayX_timeontreadmill</code>).</p></li>
<li><p><strong>VO2 Max and NMR Testing</strong>: Information from
<strong>VO2 max tests</strong> and <strong>NMR body composition
tests</strong> is included. Variables such as
<code>vo2.max.test.vo2_max</code> and <code>nmr.testing.nmr_fat</code>
capture maximum oxygen uptake, percent body fat, and other key metabolic
measures.</p></li>
<li><p><strong>Specimen Collection</strong>: Specimen collection details
are provided, including dates
(<code>specimen.collection.d_visit</code>), times of collection
(<code>specimen.processing.t_collection</code>), and types of tissues
collected
(<code>specimen.processing.sampletypedescription</code>).</p></li>
<li><p><strong>Terminal Measures</strong>: Terminal metrics, such as
body weight at sacrifice and tissue weights
(<code>terminal.weight.bw</code>, <code>terminal.weight.sol</code>),
document the final physical characteristics of the animals after
completion of the study.</p></li>
<li><p><strong>Calculated Variables</strong>: Derived variables, such as
changes in body composition
(<code>calculated.variables.pct_body_fat_change</code>), lactate levels,
and <code>vo2_max</code>, provide insights into physiological
adaptations resulting from the exercise intervention.</p></li>
<li><p><strong>Custom Variables</strong>: Variables like
<code>sacrificetime</code>, <code>intervention</code>, and
<code>time_to_freeze</code> have been added for simplified grouping and
analysis, reflecting key time points, interventions, and
time-to-processing of samples.</p></li>
</ol>
</div>
<div class="section level4">
<h4 id="key-variables-for-analysis">Key Variables for Analysis<a class="anchor" aria-label="anchor" href="#key-variables-for-analysis"></a>
</h4>
<ul>
<li>
<strong><code>pid</code></strong>: Participant Identifier. A unique
8-digit identifier assigned to each animal subject (rats in this study).
All samples (<code>viallabel</code>) coming from the same animal have
the same <code>pid</code>. This variable is crucial when combining
results from multiple assays with phenotypic data, allowing for
animal-specific longitudinal analysis.</li>
<li>
<strong><code>viallabel</code></strong>: Vial Label ID. A unique
11-digit code assigned to each sample vial. This ID is present across
all related results and metadata files, serving as a key to link the
quantitative results with the phenotypic data.</li>
<li>
<strong><code>sex</code></strong>: Sex of the animal. Represented as
<code>"female"</code> for Female and <code>"male"</code>. This variable
is critical for any analysis that aims to determine sex-based
differences.</li>
<li>
<strong><code>group</code></strong>: A simplify version of
<code>study_group_timepoint</code> that only includes the group and time
points: <code>control</code> (sedentary), and <code>1w</code>,
<code>2w</code>, <code>4w</code>, and <code>8w</code> (exercise)</li>
<li>
<strong><code>tissue</code></strong>: tissue name (short)</li>
</ul>
<p>In summary, the <strong>PHENO</strong> object provides a
comprehensive overview of each animal’s involvement in the study,
including the conditions under which they were kept, the specific
training they underwent, and the physiological changes observed as a
result of the endurance training intervention. This dataset allows for
in-depth analysis of the effects of physical activity at various
biological levels.</p>
</div>
<div class="section level4">
<h4 id="visualizations-of-phenotypes">Visualizations of Phenotypes<a class="anchor" aria-label="anchor" href="#visualizations-of-phenotypes"></a>
</h4>
<div class="section level5">
<h5 id="animal-counts">Animal counts<a class="anchor" aria-label="anchor" href="#animal-counts"></a>
</h5>
<p>First, let’s select the ids of the rats used in this study. For that,
we will first extract the samples ids, and we’ll map those samples ids
to the rats ids.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the proteomics data, for example:</span></span>
<span><span class="va">data_prot</span> <span class="op">&lt;-</span> <span class="fu">MotrpacRatTraining6mo</span><span class="fu">::</span><span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/combine_normalized_data.html" class="external-link">combine_normalized_data</a></span><span class="op">(</span></span>
<span>  assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PROT"</span>,<span class="st">"UBIQ"</span>,<span class="st">"PHOSPHO"</span>,<span class="st">"ACETYL"</span><span class="op">)</span>, </span>
<span>  exclude_outliers <span class="op">=</span> <span class="cn">TRUE</span>, <span class="op">)</span></span>
<span><span class="co">#&gt; PROT_CORTEX_NORM_DATA</span></span>
<span><span class="co">#&gt; PROT_HEART_NORM_DATA</span></span>
<span><span class="co">#&gt; PROT_KIDNEY_NORM_DATA</span></span>
<span><span class="co">#&gt; PROT_LIVER_NORM_DATA</span></span>
<span><span class="co">#&gt; PROT_LUNG_NORM_DATA</span></span>
<span><span class="co">#&gt; PROT_SKMGN_NORM_DATA</span></span>
<span><span class="co">#&gt; PROT_WATSC_NORM_DATA</span></span>
<span><span class="co">#&gt; UBIQ_HEART_NORM_DATA</span></span>
<span><span class="co">#&gt; UBIQ_LIVER_NORM_DATA</span></span>
<span><span class="co">#&gt; PHOSPHO_CORTEX_NORM_DATA</span></span>
<span><span class="co">#&gt; PHOSPHO_HEART_NORM_DATA</span></span>
<span><span class="co">#&gt; PHOSPHO_KIDNEY_NORM_DATA</span></span>
<span><span class="co">#&gt; PHOSPHO_LIVER_NORM_DATA</span></span>
<span><span class="co">#&gt; PHOSPHO_LUNG_NORM_DATA</span></span>
<span><span class="co">#&gt; PHOSPHO_SKMGN_NORM_DATA</span></span>
<span><span class="co">#&gt; PHOSPHO_WATSC_NORM_DATA</span></span>
<span><span class="co">#&gt; ACETYL_HEART_NORM_DATA</span></span>
<span><span class="co">#&gt; ACETYL_LIVER_NORM_DATA</span></span>
<span></span>
<span><span class="co"># Get the unique pid (rat id) values from the proteomics data</span></span>
<span><span class="va">rat_omics</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">data_prot</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"feature"</span>, <span class="st">"feature_ID"</span>, <span class="st">"tissue"</span>, <span class="st">"assay"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Now let’s check the total number of animals in the MoTrPAC study and
the number of animals used in the omics data.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Count unique 'pid' by 'group' and 'sex'</span></span>
<span><span class="va">count_data</span> <span class="op">&lt;-</span> <span class="va">PHENO</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">group</span>, <span class="va">sex</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>unique_pid_count <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/n_distinct.html" class="external-link">n_distinct</a></span><span class="op">(</span><span class="va">pid</span><span class="op">)</span>,</span>
<span>            .groups <span class="op">=</span> <span class="st">'drop'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">ungroup</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#  Animals used in this study</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">count_data</span>,</span>
<span>       <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">group</span>, y <span class="op">=</span> <span class="va">unique_pid_count</span>, fill <span class="op">=</span> <span class="va">sex</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>           position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html" class="external-link">position_dodge</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span>,</span>
<span>           width    <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>           color    <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="va">unique_pid_count</span><span class="op">)</span>,</span>
<span>            position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html" class="external-link">position_dodge</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span>,</span>
<span>            hjust    <span class="op">=</span> <span class="op">-</span><span class="fl">0.3</span>,</span>
<span>            size     <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>expand <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/expansion.html" class="external-link">expansion</a></span><span class="op">(</span>mult <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title    <span class="op">=</span> <span class="st">"Animal Count"</span>,</span>
<span>       subtitle <span class="op">=</span> <span class="st">"All animals processed at the Animal Site"</span>,</span>
<span>       x        <span class="op">=</span> <span class="st">"Study Group Time Point"</span>,</span>
<span>       y        <span class="op">=</span> <span class="st">"Unique Animals (pid)"</span>,</span>
<span>       fill     <span class="op">=</span> <span class="st">"Sex"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_linedraw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    plot.title       <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,</span>
<span>    plot.subtitle    <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">11</span><span class="op">)</span>,</span>
<span>    axis.text        <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    axis.title       <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">11</span><span class="op">)</span>,</span>
<span>    legend.position  <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>    legend.title     <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    panel.grid.major.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.grid.minor    <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span></span>
<span>    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>      female <span class="op">=</span> <span class="st">"#FFB6C1"</span>,</span>
<span>      male   <span class="op">=</span> <span class="st">"#1F77B4"</span><span class="op">)</span><span class="op">)</span></span>
<span>    </span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p1</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-3-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Filter by pid (i.e., the animal id)</span></span>
<span><span class="va">count_data</span> <span class="op">&lt;-</span> <span class="va">PHENO</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">rat_omics</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">group</span>, <span class="va">sex</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>unique_pid_count <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/n_distinct.html" class="external-link">n_distinct</a></span><span class="op">(</span><span class="va">pid</span><span class="op">)</span>,</span>
<span>            .groups <span class="op">=</span> <span class="st">'drop'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">ungroup</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create the grouped bar chart</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">count_data</span>,</span>
<span>       <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">group</span>, y <span class="op">=</span> <span class="va">unique_pid_count</span>, fill <span class="op">=</span> <span class="va">sex</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>           position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html" class="external-link">position_dodge</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span>,</span>
<span>           width    <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>           color    <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="va">unique_pid_count</span><span class="op">)</span>,</span>
<span>            position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html" class="external-link">position_dodge</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span>,</span>
<span>            hjust    <span class="op">=</span> <span class="op">-</span><span class="fl">0.3</span>,</span>
<span>            size     <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>expand <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/expansion.html" class="external-link">expansion</a></span><span class="op">(</span>mult <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title    <span class="op">=</span> <span class="st">"Animal Count"</span>,</span>
<span>       subtitle <span class="op">=</span> <span class="st">"Only animals used for molecular profile"</span>,</span>
<span>       x        <span class="op">=</span> <span class="st">"Study Group Time Point"</span>,</span>
<span>       y        <span class="op">=</span> <span class="st">"Unique Animals (pid)"</span>,</span>
<span>       fill     <span class="op">=</span> <span class="st">"Sex"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_linedraw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    plot.title       <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,</span>
<span>    plot.subtitle    <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">11</span><span class="op">)</span>,</span>
<span>    axis.text        <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    axis.title       <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">11</span><span class="op">)</span>,</span>
<span>    legend.position  <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>    legend.title     <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    panel.grid.major.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.grid.minor    <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span></span>
<span>    values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>      female <span class="op">=</span> <span class="st">"#FFB6C1"</span>,</span>
<span>      male   <span class="op">=</span> <span class="st">"#1F77B4"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p2</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-3-2.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level5">
<h5 id="sample-counts-by-tissue">Sample counts by tissue<a class="anchor" aria-label="anchor" href="#sample-counts-by-tissue"></a>
</h5>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Count total number of unique 'viallabel' by 'tissue'</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">viallabel_data</span> <span class="op">&lt;-</span> <span class="va">PHENO</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>    <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">tissue</span><span class="op">)</span>, <span class="co"># drop NA tissues</span></span>
<span>    <span class="va">pid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">rat_omics</span> <span class="co"># only include animals used in the omics data</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">tissue</span>, <span class="va">group</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span></span>
<span>    total_viallabel_count <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/n_distinct.html" class="external-link">n_distinct</a></span><span class="op">(</span><span class="va">viallabel</span><span class="op">)</span>,</span>
<span>    .groups <span class="op">=</span> <span class="st">"drop"</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">ungroup</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>    tissue <span class="op">=</span> <span class="fu">forcats</span><span class="fu">::</span><span class="fu"><a href="https://forcats.tidyverse.org/reference/fct_reorder.html" class="external-link">fct_reorder</a></span><span class="op">(</span><span class="va">tissue</span>, <span class="va">total_viallabel_count</span>, .fun <span class="op">=</span> <span class="va">sum</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">viallabel_data</span>,</span>
<span>       <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">tissue</span>,</span>
<span>           y <span class="op">=</span> <span class="va">total_viallabel_count</span>,</span>
<span>           fill <span class="op">=</span> <span class="va">group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_col</a></span><span class="op">(</span>position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html" class="external-link">position_dodge</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span>,</span>
<span>           width    <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>           color    <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="va">total_viallabel_count</span><span class="op">)</span>,</span>
<span>            position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html" class="external-link">position_dodge</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span>,</span>
<span>            hjust    <span class="op">=</span> <span class="op">-</span><span class="fl">0.1</span>,</span>
<span>            size     <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>expand <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/expansion.html" class="external-link">expansion</a></span><span class="op">(</span>mult <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title    <span class="op">=</span> <span class="st">"Vial-label Count by Tissue &amp; Group"</span>,</span>
<span>       x        <span class="op">=</span> <span class="st">"Tissue"</span>,</span>
<span>       y        <span class="op">=</span> <span class="st">"Distinct Vial-labels"</span>,</span>
<span>       fill     <span class="op">=</span> <span class="st">"Group"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    plot.title       <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    axis.title       <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">11</span><span class="op">)</span>,</span>
<span>    panel.grid.major.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.position    <span class="op">=</span> <span class="st">"top"</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-4-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div class="section level5">
<h5 id="weight-distributions">Weight distributions<a class="anchor" aria-label="anchor" href="#weight-distributions"></a>
</h5>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Select variables from PHENO that start with "training" and end with "_weight"</span></span>
<span><span class="va">weight_cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"^training.*_weight$"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">PHENO</span><span class="op">)</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Select those columns along with "pid", "sex", and "group"</span></span>
<span><span class="va">selected_cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pid"</span>, <span class="st">"sex"</span>, <span class="st">"group"</span>, <span class="va">weight_cols</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get unique values</span></span>
<span><span class="va">unique_pheno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">PHENO</span><span class="op">[</span>, <span class="va">selected_cols</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Reshape data to long format for plotting</span></span>
<span><span class="va">pheno_long</span> <span class="op">&lt;-</span> <span class="va">unique_pheno</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">rat_omics</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">starts_with</a></span><span class="op">(</span><span class="st">"training"</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"training_session"</span>, values_to <span class="op">=</span> <span class="st">"weight"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Define the desired order of training sessions</span></span>
<span><span class="va">training_session_order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"training.day1_weight"</span>, </span>
<span>                            <span class="st">"training.day6_weight"</span>, </span>
<span>                            <span class="st">"training.day11_weight"</span>, </span>
<span>                            <span class="st">"training.day16_weight"</span>, </span>
<span>                            <span class="st">"training.day21_weight"</span>, </span>
<span>                            <span class="st">"training.day26_weight"</span>, </span>
<span>                            <span class="st">"training.day31_weight"</span>, </span>
<span>                            <span class="st">"training.day36_weight"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert training_session to a factor with the specified order</span></span>
<span><span class="va">pheno_long</span><span class="op">$</span><span class="va">training_session</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">pheno_long</span><span class="op">$</span><span class="va">training_session</span>, </span>
<span>                                      levels <span class="op">=</span> <span class="va">training_session_order</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot the jittered plot for each training session, faceted by sex and colored by group</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">pheno_long</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">training_session</span>, y <span class="op">=</span> <span class="va">weight</span>, color <span class="op">=</span> <span class="va">group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html" class="external-link">geom_jitter</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.2</span>, size <span class="op">=</span> <span class="fl">2</span>, alpha <span class="op">=</span> <span class="fl">0.6</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">sex</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Weight by Training Session"</span>,</span>
<span>       x <span class="op">=</span> <span class="st">"Training Session"</span>,</span>
<span>       y <span class="op">=</span> <span class="st">"Weight"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p1</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-5-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">pheno_long</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">training_session</span>, y <span class="op">=</span> <span class="va">weight</span>, color <span class="op">=</span> <span class="va">group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.3</span>, outlier.shape <span class="op">=</span> <span class="cn">NA</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span>, linewidth <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">sex</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Weight by Training Session"</span>,</span>
<span>       x <span class="op">=</span> <span class="st">"Training Session"</span>,</span>
<span>       y <span class="op">=</span> <span class="st">"Weight"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p2</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-5-2.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">pheno_long</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">training_session</span>, y <span class="op">=</span> <span class="va">weight</span>, group <span class="op">=</span> <span class="va">group</span>, color <span class="op">=</span> <span class="va">group</span>, fill <span class="op">=</span> <span class="va">group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/stat_summary.html" class="external-link">stat_summary</a></span><span class="op">(</span>fun <span class="op">=</span> <span class="va">median</span>, geom <span class="op">=</span> <span class="st">"point"</span>, size <span class="op">=</span> <span class="fl">4</span>, shape <span class="op">=</span> <span class="fl">21</span>, alpha <span class="op">=</span> <span class="fl">0.7</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/stat_summary.html" class="external-link">stat_summary</a></span><span class="op">(</span>fun <span class="op">=</span> <span class="va">median</span>, geom <span class="op">=</span> <span class="st">"line"</span>, size <span class="op">=</span> <span class="fl">1</span>, alpha <span class="op">=</span> <span class="fl">0.7</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">sex</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_linedraw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Mean Weight by Training Session"</span>,</span>
<span>       x <span class="op">=</span> <span class="st">"Training Session"</span>,</span>
<span>       y <span class="op">=</span> <span class="st">"Mean Weight"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use `linewidth` instead.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p3</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-5-3.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level5">
<h5 id="vo2max">VO2max<a class="anchor" aria-label="anchor" href="#vo2max"></a>
</h5>
<p>VO2max measurements were taken at the start for all time points, and
at the end for the 4-week, 8-week, and control training sessions.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># colnames(PHENO)[grep("vo2*d_visit", colnames(PHENO))]</span></span>
<span></span>
<span><span class="co"># Select those columns along with "pid", "sex", and "group"</span></span>
<span><span class="va">selected_cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pid"</span>, </span>
<span>                   <span class="st">"sex"</span>, </span>
<span>                   <span class="st">"group"</span>, </span>
<span>                   <span class="st">"vo2.max.test.vo2_max_1"</span>, </span>
<span>                   <span class="st">"vo2.max.test.vo2_max_2"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get unique values</span></span>
<span><span class="va">unique_pheno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">PHENO</span><span class="op">[</span>, <span class="va">selected_cols</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Reshape data to long format for plotting</span></span>
<span><span class="va">pheno_long</span> <span class="op">&lt;-</span> <span class="va">unique_pheno</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">rat_omics</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">starts_with</a></span><span class="op">(</span><span class="st">"vo2"</span><span class="op">)</span>, </span>
<span>                      names_to <span class="op">=</span> <span class="st">"training_session"</span>, </span>
<span>                      values_to <span class="op">=</span> <span class="st">"vo2max"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">pheno_long</span>, </span>
<span>             <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">training_session</span>, y <span class="op">=</span> <span class="va">vo2max</span>, color <span class="op">=</span> <span class="va">group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.3</span>, outlier.shape <span class="op">=</span> <span class="cn">NA</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co"># geom_jitter(width = 0.2, size = 2, alpha = 0.6, na.rm = TRUE) +</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">sex</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"VO2max distributions at first and last training session"</span>,</span>
<span>       x <span class="op">=</span> <span class="st">"Training Session"</span>,</span>
<span>       y <span class="op">=</span> <span class="st">"VO2max"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p1</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-6-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">pheno_long</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">training_session</span>, y <span class="op">=</span> <span class="va">vo2max</span>, group <span class="op">=</span> <span class="va">group</span>, color <span class="op">=</span> <span class="va">group</span>, fill <span class="op">=</span> <span class="va">group</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/stat_summary.html" class="external-link">stat_summary</a></span><span class="op">(</span>fun <span class="op">=</span> <span class="va">mean</span>, geom <span class="op">=</span> <span class="st">"point"</span>, size <span class="op">=</span> <span class="fl">4</span>, shape <span class="op">=</span> <span class="fl">21</span>, alpha <span class="op">=</span> <span class="fl">0.7</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/stat_summary.html" class="external-link">stat_summary</a></span><span class="op">(</span>fun <span class="op">=</span> <span class="va">mean</span>, geom <span class="op">=</span> <span class="st">"line"</span>, size <span class="op">=</span> <span class="fl">1</span>, alpha <span class="op">=</span> <span class="fl">0.7</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">sex</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_linedraw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Mean VO2max at first and last training session"</span>,</span>
<span>       x <span class="op">=</span> <span class="st">"Training Session"</span>,</span>
<span>       y <span class="op">=</span> <span class="st">"Mean VO2max"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p2</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-6-2.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level5">
<h5 id="treadmill-speed-over-time">Treadmill speed over time<a class="anchor" aria-label="anchor" href="#treadmill-speed-over-time"></a>
</h5>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Filter the data for training rats</span></span>
<span><span class="va">PHENO_training</span> <span class="op">&lt;-</span> <span class="va">PHENO</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">intervention</span> <span class="op">==</span> <span class="st">"training"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Pivot the date columns</span></span>
<span><span class="va">PHENO_long_date</span> <span class="op">&lt;-</span> <span class="va">PHENO_training</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">pid</span>, <span class="va">group</span>, <span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">matches</a></span><span class="op">(</span><span class="st">"training\\.day\\d+date$"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span></span>
<span>    cols <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">pid</span>, <span class="va">group</span><span class="op">)</span>,</span>
<span>    names_to <span class="op">=</span> <span class="st">"day"</span>,</span>
<span>    names_pattern <span class="op">=</span> <span class="st">"training\\.day(\\d+)date$"</span>,</span>
<span>    values_to <span class="op">=</span> <span class="st">"date"</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>    day <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="va">day</span><span class="op">)</span>,</span>
<span>    date <span class="op">=</span> <span class="fu"><a href="https://lubridate.tidyverse.org/reference/ymd.html" class="external-link">dmy</a></span><span class="op">(</span><span class="va">date</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Pivot the treadmillspeed columns</span></span>
<span><span class="va">PHENO_long_speed</span> <span class="op">&lt;-</span> <span class="va">PHENO_training</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">pid</span>, <span class="va">group</span>, <span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">matches</a></span><span class="op">(</span><span class="st">"training\\.day\\d+_treadmillspeed$"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span></span>
<span>    cols <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">pid</span>, <span class="va">group</span><span class="op">)</span>,</span>
<span>    names_to <span class="op">=</span> <span class="st">"day"</span>,</span>
<span>    names_pattern <span class="op">=</span> <span class="st">"training\\.day(\\d+)_treadmillspeed$"</span>,</span>
<span>    values_to <span class="op">=</span> <span class="st">"treadmillspeed"</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>    day <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="va">day</span><span class="op">)</span>,</span>
<span>    treadmillspeed <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">treadmillspeed</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Merge the two datasets</span></span>
<span><span class="va">PHENO_long</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">PHENO_long_date</span>, <span class="va">PHENO_long_speed</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pid"</span>, <span class="st">"group"</span>, <span class="st">"day"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Remove rows where both 'date' and 'treadmillspeed' are NA</span></span>
<span><span class="va">PHENO_long</span> <span class="op">&lt;-</span> <span class="va">PHENO_long</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">date</span><span class="op">)</span> <span class="op">&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">treadmillspeed</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a new Id combining group and pid</span></span>
<span><span class="va">PHENO_long</span> <span class="op">&lt;-</span> <span class="va">PHENO_long</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>group_pid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">group</span>, <span class="st">"_"</span>, <span class="va">pid</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># # Identify duplicates</span></span>
<span><span class="co"># duplicates &lt;- PHENO_long %&gt;%</span></span>
<span><span class="co">#   select(group_pid, date, treadmillspeed) %&gt;%</span></span>
<span><span class="co">#   group_by(group_pid, date) %&gt;%</span></span>
<span><span class="co">#   filter(n() &gt; 1)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # View duplicates</span></span>
<span><span class="co"># print(duplicates)</span></span>
<span></span>
<span><span class="co"># Reshape the data to wide format</span></span>
<span><span class="va">PHENO_wide</span> <span class="op">&lt;-</span> <span class="va">PHENO_long</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html" class="external-link">pivot_wider</a></span><span class="op">(</span></span>
<span>    names_from <span class="op">=</span> <span class="va">date</span>,</span>
<span>    values_from <span class="op">=</span> <span class="va">treadmillspeed</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Aggregate duplicates by taking the mean of treadmillspeed</span></span>
<span><span class="va">PHENO_long</span> <span class="op">&lt;-</span> <span class="va">PHENO_long</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">group_pid</span>, <span class="va">date</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span></span>
<span>    treadmillspeed <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">treadmillspeed</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>    .groups <span class="op">=</span> <span class="st">'drop'</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Reshape the data to wide format</span></span>
<span><span class="va">PHENO_wide</span> <span class="op">&lt;-</span> <span class="va">PHENO_long</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html" class="external-link">pivot_wider</a></span><span class="op">(</span></span>
<span>    names_from <span class="op">=</span> <span class="va">date</span>,</span>
<span>    values_from <span class="op">=</span> <span class="va">treadmillspeed</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert the data frame to a matrix</span></span>
<span><span class="va">PHENO_matrix</span> <span class="op">&lt;-</span> <span class="va">PHENO_wide</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"group_pid"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Replace NA values with zeros</span></span>
<span><span class="va">PHENO_matrix</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">PHENO_matrix</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0</span></span>
<span></span>
<span><span class="co"># Create the heatmap</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span></span>
<span>  <span class="va">PHENO_matrix</span>,</span>
<span>  cluster_rows <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  cluster_cols <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  show_rownames <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  show_colnames <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  scale <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>  fontsize_row <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  fontsize_col <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  angle_col <span class="op">=</span> <span class="fl">45</span>,</span>
<span>  color <span class="op">=</span> <span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">viridis</a></span><span class="op">(</span><span class="fl">100</span><span class="op">)</span>,</span>
<span>  main <span class="op">=</span> <span class="st">"Treadmill Speed Over Time for Training Rats"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-7-1.png" width="768" style="display: block; margin: auto;"></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="data">DATA<a class="anchor" aria-label="anchor" href="#data"></a>
</h2>
<p><strong>Tip:</strong> To learn more about any data object, use
<code>?</code> to retrieve the documentation, e.g.,
<code><a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/METAB_FEATURE_ID_MAP.html" class="external-link">?METAB_FEATURE_ID_MAP</a></code>. Note that
<code>MotrpacRatTraining6moData</code> must be installed and loaded with
<code><a href="https://rdrr.io/r/base/library.html" class="external-link">library()</a></code> for this to work.</p>
<div class="section level3">
<h3 id="tissue-and-assay-abbreviations">Tissue and assay abbreviations<a class="anchor" aria-label="anchor" href="#tissue-and-assay-abbreviations"></a>
</h3>
<p>It is important to be aware of the tissue and assay abbreviations
because they are used to name many data objects. The vectors of
abbreviations are also available in <code>TISSUE_ABBREV</code> and
<code>ASSAY_ABBREV</code>.</p>
<div class="section level4">
<h4 id="tissues">Tissues<a class="anchor" aria-label="anchor" href="#tissues"></a>
</h4>
<ul>
<li>
<strong>ADRNL:</strong> adrenal gland</li>
<li>
<strong>BAT:</strong> brown adipose tissue</li>
<li>
<strong>BLOOD:</strong> whole blood</li>
<li>
<strong>COLON:</strong> colon</li>
<li>
<strong>CORTEX:</strong> cerebral cortex</li>
<li>
<strong>HEART:</strong> heart</li>
<li>
<strong>HIPPOC:</strong> hippocampus</li>
<li>
<strong>HYPOTH:</strong> hypothalamus</li>
<li>
<strong>KIDNEY:</strong> kidney</li>
<li>
<strong>LIVER:</strong> liver</li>
<li>
<strong>LUNG:</strong> lung</li>
<li>
<strong>OVARY:</strong> ovaries (female gonads)</li>
<li>
<strong>PLASMA:</strong> plasma from blood</li>
<li>
<strong>SKM-GN:</strong> gastrocnemius (leg skeletal muscle)</li>
<li>
<strong>SKM-VL:</strong> vastus lateralis (leg skeletal muscle)</li>
<li>
<strong>SMLINT:</strong> small intestine</li>
<li>
<strong>SPLEEN:</strong> spleen</li>
<li>
<strong>TESTES:</strong> testes (male gonads)</li>
<li>
<strong>VENACV:</strong> vena cava</li>
<li>
<strong>WAT-SC:</strong> subcutaneous white adipose tissue</li>
</ul>
</div>
<div class="section level4">
<h4 id="assaysomes">Assays/omes<a class="anchor" aria-label="anchor" href="#assaysomes"></a>
</h4>
<ul>
<li>
<strong>ACETYL:</strong> acetylproteomics; protein site
acetylation</li>
<li>
<strong>ATAC:</strong> chromatin accessibility, ATAC-seq data</li>
<li>
<strong>IMMUNO:</strong> multiplexed immunoassays (cytokines and
hormones)</li>
<li>
<strong>METAB:</strong> metabolomics and lipidomics</li>
<li>
<strong>METHYL:</strong> DNA methylation, RRBS data</li>
<li>
<strong>PHOSPHO:</strong> phosphoproteomics; protein site
phosphorylation</li>
<li>
<strong>PROT:</strong> global proteomics; protein abundance</li>
<li>
<strong>TRNSCRPT:</strong> transcriptomics, RNA-Seq data</li>
<li>
<strong>UBIQ:</strong> ubiquitylome; protein site
ubiquitination</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="summary-of-data">Summary of data<a class="anchor" aria-label="anchor" href="#summary-of-data"></a>
</h3>
<p>Here is a brief summary of the kinds of data included in the data
package:</p>
<ul>
<li>Assay, tissue, sex, and training group abbreviations, codes, colors,
and order used in plots</li>
<li>Phenotypic data, <code>PHENO</code>
</li>
<li>Mapping between various feature identifiers, i.e.,
<code>FEATURE_TO_GENE</code>, <code>RAT_TO_HUMAN_GENE</code>
</li>
<li>Ome-specific feature annotation, i.e.,
<code>METAB_FEATURE_ID_MAP</code>, <code>METHYL_FEATURE_ANNOT</code>
(GCP only), <code>ATAC_FEATURE_ANNOT</code> (GCP only),
<code>PROT_FEATURE_ANNOT</code> (GCP only),
<code>PHOSPHO_FEATURE_ANNOT</code> (GCP only),
<code>UBIQ_FEATURE_ANNOT</code> (GCP only),
<code>ACETYL_FEATURE_ANNOT</code> (GCP only),
<code>TRNSCRPT_FEATURE_ANNOT</code> (GCP only)</li>
<li>Ome-specific sample-level metadata, i.e.,
<code>TRNSCRPT_META</code>, <code>ATAC_META</code>,
<code>METHYL_META</code>, <code>IMMUNO_META</code>,
<code>PHOSPHO_META</code>, <code>PROT_META</code>,
<code>ACETYL_META</code>, <code>UBIQ_META</code>
</li>
<li>Raw counts for RNA-Seq (TRNSCRPT), ATAC-Seq (ATAC), and RRBS
(METHYL) data, e.g., <code>TRNSCRPT_LIVER_RAW_COUNTS</code>. Note that
epigenetic data (ATAC and METHYL) must be downloaded from Google Cloud
Storage. See more details <a href="https://motrpac.github.io/MotrpacRatTraining6moData/index.html#access-epigenomics-data-through-google-cloud-storage" class="external-link">here</a>.</li>
<li>Normalized sample-level data, e.g.,
<code>TRNSCRPT_SKMGN_NORM_DATA</code>
</li>
<li>Differential analysis results, e.g., <code>HEART_PROT_DA</code>
</li>
<li>Sample outliers excluded from differential analysis,
<code>OUTLIERS</code>
</li>
<li>Table of training-regulated features at 5% FDR,
<code>TRAINING_REGULATED_FEATURES</code>
</li>
<li>Bayesian graphical analysis inputs and results</li>
<li>Pathway enrichment of main graphical clusters,
<code>GRAPH_PW_ENRICH</code>
</li>
</ul>
<p>A list of all the available data objects and a brief description are
available <a href="https://motrpac.github.io/MotrpacRatTraining6moData/index.html#explore-available-data-objects" class="external-link">here</a>.</p>
<div class="section level4">
<h4 id="samples-used-in-this-study">Samples used in this study<a class="anchor" aria-label="anchor" href="#samples-used-in-this-study"></a>
</h4>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># List all data objects in the package</span></span>
<span><span class="va">all_objects</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"MotrpacRatTraining6moData"</span><span class="op">)</span><span class="op">$</span><span class="va">results</span><span class="op">[</span>, <span class="st">"Item"</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Filter the objects that contain the string "NORM" and exclude specific unwanted objects</span></span>
<span><span class="va">exclude_objects</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"IMMUNO_NORM_DATA_FLAT"</span>, <span class="st">"IMMUNO_NORM_DATA_NESTED"</span>, </span>
<span>                     <span class="st">"METAB_NORM_DATA_FLAT"</span>, <span class="st">"METAB_NORM_DATA_NESTED"</span>, </span>
<span>                     <span class="st">"TRAINING_REGULATED_NORM_DATA"</span>, <span class="st">"TRAINING_REGULATED_NORM_DATA_NO_OUTLIERS"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">norm_objects</span> <span class="op">&lt;-</span> <span class="va">all_objects</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"NORM_DATA"</span>, <span class="va">all_objects</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">!</span><span class="op">(</span><span class="va">all_objects</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">exclude_objects</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="metabolomics-summaries">Metabolomics summaries<a class="anchor" aria-label="anchor" href="#metabolomics-summaries"></a>
</h3>
<p>MoTrPAC run multiple Metabolomics assays, both Targeted and
Untargeted. Let’s summarize the data and visualize the metabolite counts
across all tissues and datasets.</p>
<div class="section level5">
<h5 id="metabolite-count">Metabolite count<a class="anchor" aria-label="anchor" href="#metabolite-count"></a>
</h5>
<p>Individual platforms</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate the unique feature counts per dataset</span></span>
<span><span class="va">feature_counts</span> <span class="op">&lt;-</span> <span class="va">METAB_NORM_DATA_FLAT</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">dataset</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>  <span class="co"># Group only by dataset</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>unique_feature_count <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/n_distinct.html" class="external-link">n_distinct</a></span><span class="op">(</span><span class="va">feature_ID</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">unique_feature_count</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create the bar plot</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">feature_counts</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/reorder.factor.html" class="external-link">reorder</a></span><span class="op">(</span><span class="va">dataset</span>, <span class="va">unique_feature_count</span><span class="op">)</span>, y <span class="op">=</span> <span class="va">unique_feature_count</span>, fill <span class="op">=</span> <span class="va">dataset</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="va">unique_feature_count</span><span class="op">)</span>, </span>
<span>            hjust <span class="op">=</span> <span class="op">-</span><span class="fl">0.1</span>,  </span>
<span>            size <span class="op">=</span> <span class="fl">3.5</span><span class="op">)</span> <span class="op">+</span>  </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Total number of Metabolites by Platform"</span>,</span>
<span>       caption <span class="op">=</span> <span class="st">"Across all tissues"</span>,</span>
<span>       x <span class="op">=</span> <span class="st">""</span>,</span>
<span>       y <span class="op">=</span> <span class="st">"Metabolite count"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>  <span class="co"># Flip coordinates for better readability</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>        legend.position <span class="op">=</span> <span class="st">"none"</span>,  <span class="co"># Hide legend for dataset colors</span></span>
<span>        plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span>, face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Print the plot</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-9-1.png" width="700" style="display: block; margin: auto;"></p>
<p>Combining all platforms</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Add metab_type based on dataset</span></span>
<span><span class="va">METAB_NORM_DATA_FLAT</span> <span class="op">&lt;-</span> <span class="va">METAB_NORM_DATA_FLAT</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>    metab_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"-t-"</span>, <span class="va">dataset</span><span class="op">)</span>, <span class="st">"targeted"</span>, <span class="st">"untargeted"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Count the number of feature_ID for each dataset, tissue, and metab_type</span></span>
<span><span class="va">feature_counts</span> <span class="op">&lt;-</span> <span class="va">METAB_NORM_DATA_FLAT</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">dataset</span>, <span class="va">tissue</span>, <span class="va">metab_type</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>feature_count <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/n_distinct.html" class="external-link">n_distinct</a></span><span class="op">(</span><span class="va">feature_ID</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot the results with facets for metab_type</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">feature_counts</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">tissue</span>, y <span class="op">=</span> <span class="va">feature_count</span>, fill <span class="op">=</span> <span class="va">dataset</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>    title <span class="op">=</span> <span class="st">"Number of metabolites by Dataset, Tissue, and Metab Type"</span>,</span>
<span>    x <span class="op">=</span> <span class="st">"Tissue"</span>,</span>
<span>    y <span class="op">=</span> <span class="st">"Feature Count"</span>,</span>
<span>    fill <span class="op">=</span> <span class="st">"Dataset"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">metab_type</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-10-1.png" width="700" style="display: block; margin: auto;"></p>
<p>Details by Targeted and Untargeted</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">feature_counts_complete</span> <span class="op">&lt;-</span> <span class="va">feature_counts</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/complete.html" class="external-link">complete</a></span><span class="op">(</span><span class="va">tissue</span>, <span class="va">dataset</span>, <span class="va">metab_type</span>, fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>feature_count <span class="op">=</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Loop over metab_type and create a plot for each</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">type</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">feature_counts_complete</span><span class="op">$</span><span class="va">metab_type</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Filter for the current metab_type and its datasets</span></span>
<span>  <span class="va">type_data</span> <span class="op">&lt;-</span> <span class="va">feature_counts_complete</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">metab_type</span> <span class="op">==</span> <span class="va">type</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="kw">if</span> <span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"targeted"</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"-t-"</span>, <span class="va">dataset</span><span class="op">)</span> <span class="kw">else</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"-t-"</span>, <span class="va">dataset</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Create the plot</span></span>
<span>  <span class="va">plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">type_data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">tissue</span>, y <span class="op">=</span> <span class="va">feature_count</span>, fill <span class="op">=</span> <span class="va">dataset</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html" class="external-link">position_dodge</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>      title <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Metabolite Count by Dataset and Tissue ("</span>, <span class="va">type</span>, <span class="st">")"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>,</span>
<span>      x <span class="op">=</span> <span class="st">"Tissue"</span>,</span>
<span>      y <span class="op">=</span> <span class="st">"Metabolite Count"</span>,</span>
<span>      fill <span class="op">=</span> <span class="st">"Metabolomics Assay"</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>      axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>      axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>      axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,</span>
<span>      axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,</span>
<span>      plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span>, face <span class="op">=</span> <span class="st">"bold"</span>, hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>      panel.grid.major.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>      panel.grid.major.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_line</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"gray"</span>, linetype <span class="op">=</span> <span class="st">"dotted"</span><span class="op">)</span>,</span>
<span>      panel.border <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_rect</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"black"</span>, fill <span class="op">=</span> <span class="cn">NA</span>, size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_discrete.html" class="external-link">scale_x_discrete</a></span><span class="op">(</span>expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.05</span>, <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set3"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Print the plot</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">plot</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="co">#&gt; Warning: The `size` argument of `element_rect()` is deprecated as of ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use the `linewidth` argument instead.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-11-1.png" width="768" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-11-2.png" width="768" style="display: block; margin: auto;">
Total counts by tissue</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Ensure all combinations of tissue and dataset exist, but separately for each metab_type</span></span>
<span><span class="va">feature_counts_complete</span> <span class="op">&lt;-</span> <span class="va">feature_counts</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/complete.html" class="external-link">complete</a></span><span class="op">(</span><span class="va">tissue</span>, <span class="va">dataset</span>, <span class="va">metab_type</span>, fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>feature_count <span class="op">=</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Loop over metab_type and create a plot for each</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">type</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">feature_counts_complete</span><span class="op">$</span><span class="va">metab_type</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Filter for the current metab_type and its datasets</span></span>
<span>  <span class="va">type_data</span> <span class="op">&lt;-</span> <span class="va">feature_counts_complete</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">metab_type</span> <span class="op">==</span> <span class="va">type</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="kw">if</span> <span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"targeted"</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"-t-"</span>, <span class="va">dataset</span><span class="op">)</span> <span class="kw">else</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"-t-"</span>, <span class="va">dataset</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Create the plot</span></span>
<span>  <span class="va">plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">type_data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">tissue</span>, y <span class="op">=</span> <span class="va">feature_count</span>, fill <span class="op">=</span> <span class="va">dataset</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"stack"</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>      title <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Metabolite Count by Dataset and Tissue ("</span>, <span class="va">type</span>, <span class="st">")"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>,</span>
<span>      x <span class="op">=</span> <span class="st">"Tissue"</span>,</span>
<span>      y <span class="op">=</span> <span class="st">"Metabolite Count"</span>,</span>
<span>      fill <span class="op">=</span> <span class="st">"Metabolomics Assay"</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>      axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>      axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>      axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,</span>
<span>      axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,</span>
<span>      plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span>, face <span class="op">=</span> <span class="st">"bold"</span>, hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>      panel.grid.major.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>      panel.grid.major.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_line</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"gray"</span>, linetype <span class="op">=</span> <span class="st">"dotted"</span><span class="op">)</span>,</span>
<span>      panel.border <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_rect</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"black"</span>, fill <span class="op">=</span> <span class="cn">NA</span>, size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_discrete.html" class="external-link">scale_x_discrete</a></span><span class="op">(</span>expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.05</span>, <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set3"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Print the plot</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">plot</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-12-1.png" width="864" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-12-2.png" width="864" style="display: block; margin: auto;">
Total counts by targeted and untargeted</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Summarize the feature counts for each tissue and metab_type</span></span>
<span><span class="va">feature_counts_summary</span> <span class="op">&lt;-</span> <span class="va">feature_counts</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">tissue</span>, <span class="va">metab_type</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>total_features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">feature_count</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create the combined stacked bar plot</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">feature_counts_summary</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">tissue</span>, y <span class="op">=</span> <span class="va">total_features</span>, fill <span class="op">=</span> <span class="va">metab_type</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"stack"</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_light</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>    title <span class="op">=</span> <span class="st">"Metabolite Counts by Tissue (Targeted &amp; Untargeted)"</span>,</span>
<span>    x <span class="op">=</span> <span class="st">"Tissue"</span>,</span>
<span>    y <span class="op">=</span> <span class="st">"Total Feature Count"</span>,</span>
<span>    fill <span class="op">=</span> <span class="st">"Metabolomics Type"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span>, vjust <span class="op">=</span> <span class="fl">1</span>, size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>,</span>
<span>    axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,</span>
<span>    plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span>, face <span class="op">=</span> <span class="st">"bold"</span>, hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>    panel.grid.major.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.grid.major.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_line</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"gray"</span>, linetype <span class="op">=</span> <span class="st">"dotted"</span><span class="op">)</span>,</span>
<span>    panel.border <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_rect</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"black"</span>, fill <span class="op">=</span> <span class="cn">NA</span>, size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"right"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"targeted"</span> <span class="op">=</span> <span class="st">"steelblue"</span>, <span class="st">"untargeted"</span> <span class="op">=</span> <span class="st">"coral"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-13-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level5">
<h5 id="metabolite-overlap-between-features">Metabolite Overlap between features<a class="anchor" aria-label="anchor" href="#metabolite-overlap-between-features"></a>
</h5>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://github.com/hms-dbmi/UpSetR" class="external-link">UpSetR</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Prepare the data for the UpSet plot</span></span>
<span><span class="co"># Ensure there is one entry per feature_ID and dataset</span></span>
<span><span class="va">upset_data</span> <span class="op">&lt;-</span> <span class="va">METAB_NORM_DATA_FLAT</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">feature_ID</span>, <span class="va">dataset</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>  <span class="co"># Remove duplicates</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>value <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>  <span class="co"># Add a value column to indicate presence</span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html" class="external-link">pivot_wider</a></span><span class="op">(</span>names_from <span class="op">=</span> <span class="va">dataset</span>, values_from <span class="op">=</span> <span class="va">value</span>, values_fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>value <span class="op">=</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert to the format suitable for UpSetR</span></span>
<span><span class="va">upset_input</span> <span class="op">&lt;-</span> <span class="va">upset_data</span><span class="op">[</span>, <span class="op">-</span><span class="fl">1</span><span class="op">]</span>  <span class="co"># Remove the feature_ID column</span></span>
<span></span>
<span><span class="co"># Generate the UpSet plot</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/UpSetR/man/upset.html" class="external-link">upset</a></span><span class="op">(</span></span>
<span>  <span class="va">upset_input</span>,</span>
<span>  sets <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">upset_input</span><span class="op">)</span>,</span>
<span>  sets.bar.color <span class="op">=</span> <span class="st">"skyblue"</span>,</span>
<span>  main.bar.color <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>  matrix.color <span class="op">=</span> <span class="st">"blue"</span>,</span>
<span>  text.scale <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1.5</span>, <span class="fl">1</span>, <span class="fl">1.5</span><span class="op">)</span>,</span>
<span>  order.by <span class="op">=</span> <span class="st">"freq"</span>,</span>
<span>  keep.order <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-14-1.png" width="864" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level3">
<h3 id="load-the-data">Load the data<a class="anchor" aria-label="anchor" href="#load-the-data"></a>
</h3>
<p>Use <code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/load_sample_data.html" class="external-link">load_sample_data()</a></code> to load sample-level data for a
specific ome and tissue. Here we fetch various forms of the sample-level
RNA-Seq (TRNSCRPT) data for brown adipose tissue (BAT) as an
example.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load RNA-seq raw counts for brown adipose tissue </span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/load_sample_data.html" class="external-link">load_sample_data</a></span><span class="op">(</span><span class="st">"BAT"</span>, <span class="st">"TRNSCRPT"</span>, normalized <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load the normalized RNA-seq data for brown adipose tissue instead</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/load_sample_data.html" class="external-link">load_sample_data</a></span><span class="op">(</span><span class="st">"BAT"</span>, <span class="st">"TRNSCRPT"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load the normalized RNA-seq data for brown adipose tissue, but exclude sample outliers </span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/load_sample_data.html" class="external-link">load_sample_data</a></span><span class="op">(</span><span class="st">"BAT"</span>, <span class="st">"TRNSCRPT"</span>, exclude_outliers <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load the normalized RNA-seq data for brown adipose tissue for training-regulated features only</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/load_sample_data.html" class="external-link">load_sample_data</a></span><span class="op">(</span><span class="st">"BAT"</span>, <span class="st">"TRNSCRPT"</span>, training_regulated_only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/load_sample_data.html" class="external-link">load_sample_data()</a></code> will tell you if the specified
dataset doesn’t exist.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">=</span> <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/load_sample_data.html" class="external-link">load_sample_data</a></span><span class="op">(</span><span class="st">"VENACV"</span>, <span class="st">"PROT"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in load_sample_data("VENACV", "PROT"): No data returned for tissue</span></span>
<span><span class="co">#&gt; VENACV and assay PROT with current arguments.</span></span></code></pre></div>
<p><code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/load_sample_data.html" class="external-link">load_sample_data()</a></code> will also download epigenetic data
from Google Cloud Storage.</p>
<div class="yellow">
<p><strong>Note:</strong> Epigenetic data require substantially more
memory than other omes.</p>
</div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load ATAC-seq raw counts for hippocampus, excluding outliers </span></span>
<span><span class="va">data</span> <span class="op">=</span> <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/load_sample_data.html" class="external-link">load_sample_data</a></span><span class="op">(</span><span class="st">"HIPPOC"</span>, </span>
<span>                        <span class="st">"ATAC"</span>, </span>
<span>                        exclude_outliers <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                        normalized <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                        scratchdir <span class="op">=</span> <span class="st">"/tmp"</span><span class="op">)</span></span></code></pre></div>
<p>We can plot the normalized sample-level data for a single feature
using <code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/plot_feature_normalized_data.html" class="external-link">plot_feature_normalized_data()</a></code>. All of the following
examples are different ways to plot the same feature.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/plot_feature_normalized_data.html" class="external-link">plot_feature_normalized_data</a></span><span class="op">(</span>feature <span class="op">=</span> <span class="st">"PROT;SKM-GN;NP_786937.1"</span>,</span>
<span>                             add_gene_symbol <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/plot_feature_normalized_data-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Two different ways to plot the same results for one feature</span></span>
<span><span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/plot_feature_normalized_data.html" class="external-link">plot_feature_normalized_data</a></span><span class="op">(</span>feature <span class="op">=</span> <span class="st">"PROT;SKM-GN;NP_786937.1"</span>,</span>
<span>                             exclude_outliers <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                             scale_x_by_time <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/plot_feature_normalized_data-2.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/plot_feature_normalized_data.html" class="external-link">plot_feature_normalized_data</a></span><span class="op">(</span>assay <span class="op">=</span> <span class="st">"PROT"</span>,</span>
<span>                             tissue <span class="op">=</span> <span class="st">"SKM-GN"</span>,</span>
<span>                             feature_ID <span class="op">=</span> <span class="st">"NP_786937.1"</span>,</span>
<span>                             exclude_outliers <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                             scale_x_by_time <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/plot_feature_normalized_data-3.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/plot_feature_normalized_data.html" class="external-link">plot_feature_normalized_data</a></span><span class="op">(</span>assay <span class="op">=</span> <span class="st">"PROT"</span>,</span>
<span>                             tissue <span class="op">=</span> <span class="st">"SKM-GN"</span>,</span>
<span>                             feature_ID <span class="op">=</span> <span class="st">"NP_786937.1"</span>,</span>
<span>                             facet_by_sex <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/plot_feature_normalized_data-4.png" width="700" style="display: block; margin: auto;"></p>
<p><code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/combine_normalized_data.html" class="external-link">combine_normalized_data()</a></code> is a wrapper for
<code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/load_sample_data.html" class="external-link">load_sample_data()</a></code> that returns combined sample-level
normalized data for multiple datasets. Note that the sample-specific
vial labels used as column names for most of the sample-level data are
replaced with rat-specific participant IDs (PIDs) to allow measurements
from multiple datasets for the same animal to be concatenated.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Return all normalized RNA-seq data</span></span>
<span><span class="va">data_tra</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/combine_normalized_data.html" class="external-link">combine_normalized_data</a></span><span class="op">(</span>assays <span class="op">=</span> <span class="st">"TRNSCRPT"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Return all normalized proteomics data. Exclude outliers </span></span>
<span><span class="va">data_prot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/combine_normalized_data.html" class="external-link">combine_normalized_data</a></span><span class="op">(</span>assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PROT"</span>,<span class="st">"UBIQ"</span>,<span class="st">"PHOSPHO"</span>,<span class="st">"ACETYL"</span><span class="op">)</span>,</span>
<span>                               exclude_outliers <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ratid_tra</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">data_tra</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"feature"</span>, <span class="st">"feature_ID"</span>, <span class="st">"tissue"</span>, <span class="st">"assay"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">ratid_prot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">data_prot</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"feature"</span>, <span class="st">"feature_ID"</span>, <span class="st">"tissue"</span>, <span class="st">"assay"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Return normalized ATAC-seq data for training-regulated features </span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/combine_normalized_data.html" class="external-link">combine_normalized_data</a></span><span class="op">(</span>assays <span class="op">=</span> <span class="st">"ATAC"</span>, </span>
<span>                               training_regulated_only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Return all non-epigenetic data</span></span>
<span><span class="co"># Note that the "include_epigen" argument is FALSE by default</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/combine_normalized_data.html" class="external-link">combine_normalized_data</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Similarly, <code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/combine_da_results.html" class="external-link">combine_da_results()</a></code> concatenates
differential analysis results for multiple datasets.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Return all non-epigenetic differential analysis results, </span></span>
<span><span class="co"># including meta-regression results for metabolomics</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/combine_da_results.html" class="external-link">combine_da_results</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Return METHYL and ATAC differential analysis results for gastrocnemius </span></span>
<span><span class="co"># res = combine_da_results(tissues="SKM-GN", </span></span>
<span><span class="co">#                          assays=c("ATAC","METHYL"),</span></span>
<span><span class="co">#                          include_epigen=TRUE)</span></span></code></pre></div>
<p>Finally, <code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/list_available_data.html" class="external-link">list_available_data()</a></code> returns a list of all of
the available data objects in the specified package.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/list_available_data.html" class="external-link">list_available_data</a></span><span class="op">(</span><span class="st">"MotrpacRatTraining6moData"</span><span class="op">)</span></span>
<span><span class="co">#&gt;   [1] "ACETYL_HEART_DA"                         </span></span>
<span><span class="co">#&gt;   [2] "ACETYL_HEART_NORM_DATA"                  </span></span>
<span><span class="co">#&gt;   [3] "ACETYL_LIVER_DA"                         </span></span>
<span><span class="co">#&gt;   [4] "ACETYL_LIVER_NORM_DATA"                  </span></span>
<span><span class="co">#&gt;   [5] "ACETYL_META"                             </span></span>
<span><span class="co">#&gt;   [6] "ASSAY_ABBREV"                            </span></span>
<span><span class="co">#&gt;   [7] "ASSAY_ABBREV_TO_CODE"                    </span></span>
<span><span class="co">#&gt;   [8] "ASSAY_CODE_TO_ABBREV"                    </span></span>
<span><span class="co">#&gt;   [9] "ASSAY_COLORS"                            </span></span>
<span><span class="co">#&gt;  [10] "ASSAY_ORDER"                             </span></span>
<span><span class="co">#&gt;  [11] "ATAC_BAT_NORM_DATA_05FDR"                </span></span>
<span><span class="co">#&gt;  [12] "ATAC_HEART_NORM_DATA_05FDR"              </span></span>
<span><span class="co">#&gt;  [13] "ATAC_HIPPOC_NORM_DATA_05FDR"             </span></span>
<span><span class="co">#&gt;  [14] "ATAC_KIDNEY_NORM_DATA_05FDR"             </span></span>
<span><span class="co">#&gt;  [15] "ATAC_LIVER_NORM_DATA_05FDR"              </span></span>
<span><span class="co">#&gt;  [16] "ATAC_LUNG_NORM_DATA_05FDR"               </span></span>
<span><span class="co">#&gt;  [17] "ATAC_META"                               </span></span>
<span><span class="co">#&gt;  [18] "ATAC_SKMGN_NORM_DATA_05FDR"              </span></span>
<span><span class="co">#&gt;  [19] "ATAC_WATSC_NORM_DATA_05FDR"              </span></span>
<span><span class="co">#&gt;  [20] "FEATURE_TO_GENE"                         </span></span>
<span><span class="co">#&gt;  [21] "FEATURE_TO_GENE_FILT"                    </span></span>
<span><span class="co">#&gt;  [22] "GENE_UNIVERSES"                          </span></span>
<span><span class="co">#&gt;  [23] "GRAPH_COMPONENTS"                        </span></span>
<span><span class="co">#&gt;  [24] "GRAPH_PW_ENRICH"                         </span></span>
<span><span class="co">#&gt;  [25] "GRAPH_STATES"                            </span></span>
<span><span class="co">#&gt;  [26] "GROUP_COLORS"                            </span></span>
<span><span class="co">#&gt;  [27] "IMMUNO_ADRNL_DA"                         </span></span>
<span><span class="co">#&gt;  [28] "IMMUNO_BAT_DA"                           </span></span>
<span><span class="co">#&gt;  [29] "IMMUNO_COLON_DA"                         </span></span>
<span><span class="co">#&gt;  [30] "IMMUNO_CORTEX_DA"                        </span></span>
<span><span class="co">#&gt;  [31] "IMMUNO_HEART_DA"                         </span></span>
<span><span class="co">#&gt;  [32] "IMMUNO_HIPPOC_DA"                        </span></span>
<span><span class="co">#&gt;  [33] "IMMUNO_KIDNEY_DA"                        </span></span>
<span><span class="co">#&gt;  [34] "IMMUNO_LIVER_DA"                         </span></span>
<span><span class="co">#&gt;  [35] "IMMUNO_LUNG_DA"                          </span></span>
<span><span class="co">#&gt;  [36] "IMMUNO_META"                             </span></span>
<span><span class="co">#&gt;  [37] "IMMUNO_NORM_DATA_FLAT"                   </span></span>
<span><span class="co">#&gt;  [38] "IMMUNO_NORM_DATA_NESTED"                 </span></span>
<span><span class="co">#&gt;  [39] "IMMUNO_OVARY_DA"                         </span></span>
<span><span class="co">#&gt;  [40] "IMMUNO_PLASMA_DA"                        </span></span>
<span><span class="co">#&gt;  [41] "IMMUNO_SKMGN_DA"                         </span></span>
<span><span class="co">#&gt;  [42] "IMMUNO_SKMVL_DA"                         </span></span>
<span><span class="co">#&gt;  [43] "IMMUNO_SMLINT_DA"                        </span></span>
<span><span class="co">#&gt;  [44] "IMMUNO_SPLEEN_DA"                        </span></span>
<span><span class="co">#&gt;  [45] "IMMUNO_TESTES_DA"                        </span></span>
<span><span class="co">#&gt;  [46] "IMMUNO_WATSC_DA"                         </span></span>
<span><span class="co">#&gt;  [47] "METAB_ADRNL_DA"                          </span></span>
<span><span class="co">#&gt;  [48] "METAB_ADRNL_DA_METAREG"                  </span></span>
<span><span class="co">#&gt;  [49] "METAB_BAT_DA"                            </span></span>
<span><span class="co">#&gt;  [50] "METAB_BAT_DA_METAREG"                    </span></span>
<span><span class="co">#&gt;  [51] "METAB_COLON_DA"                          </span></span>
<span><span class="co">#&gt;  [52] "METAB_COLON_DA_METAREG"                  </span></span>
<span><span class="co">#&gt;  [53] "METAB_CORTEX_DA"                         </span></span>
<span><span class="co">#&gt;  [54] "METAB_CORTEX_DA_METAREG"                 </span></span>
<span><span class="co">#&gt;  [55] "METAB_FEATURE_ID_MAP"                    </span></span>
<span><span class="co">#&gt;  [56] "METAB_HEART_DA"                          </span></span>
<span><span class="co">#&gt;  [57] "METAB_HEART_DA_METAREG"                  </span></span>
<span><span class="co">#&gt;  [58] "METAB_HIPPOC_DA"                         </span></span>
<span><span class="co">#&gt;  [59] "METAB_HIPPOC_DA_METAREG"                 </span></span>
<span><span class="co">#&gt;  [60] "METAB_HYPOTH_DA"                         </span></span>
<span><span class="co">#&gt;  [61] "METAB_HYPOTH_DA_METAREG"                 </span></span>
<span><span class="co">#&gt;  [62] "METAB_KIDNEY_DA"                         </span></span>
<span><span class="co">#&gt;  [63] "METAB_KIDNEY_DA_METAREG"                 </span></span>
<span><span class="co">#&gt;  [64] "METAB_LIVER_DA"                          </span></span>
<span><span class="co">#&gt;  [65] "METAB_LIVER_DA_METAREG"                  </span></span>
<span><span class="co">#&gt;  [66] "METAB_LUNG_DA"                           </span></span>
<span><span class="co">#&gt;  [67] "METAB_LUNG_DA_METAREG"                   </span></span>
<span><span class="co">#&gt;  [68] "METAB_NORM_DATA_FLAT"                    </span></span>
<span><span class="co">#&gt;  [69] "METAB_NORM_DATA_NESTED"                  </span></span>
<span><span class="co">#&gt;  [70] "METAB_OVARY_DA"                          </span></span>
<span><span class="co">#&gt;  [71] "METAB_OVARY_DA_METAREG"                  </span></span>
<span><span class="co">#&gt;  [72] "METAB_PLASMA_DA"                         </span></span>
<span><span class="co">#&gt;  [73] "METAB_PLASMA_DA_METAREG"                 </span></span>
<span><span class="co">#&gt;  [74] "METAB_SKMGN_DA"                          </span></span>
<span><span class="co">#&gt;  [75] "METAB_SKMGN_DA_METAREG"                  </span></span>
<span><span class="co">#&gt;  [76] "METAB_SKMVL_DA"                          </span></span>
<span><span class="co">#&gt;  [77] "METAB_SKMVL_DA_METAREG"                  </span></span>
<span><span class="co">#&gt;  [78] "METAB_SMLINT_DA"                         </span></span>
<span><span class="co">#&gt;  [79] "METAB_SMLINT_DA_METAREG"                 </span></span>
<span><span class="co">#&gt;  [80] "METAB_SPLEEN_DA"                         </span></span>
<span><span class="co">#&gt;  [81] "METAB_SPLEEN_DA_METAREG"                 </span></span>
<span><span class="co">#&gt;  [82] "METAB_TESTES_DA"                         </span></span>
<span><span class="co">#&gt;  [83] "METAB_TESTES_DA_METAREG"                 </span></span>
<span><span class="co">#&gt;  [84] "METAB_VENACV_DA"                         </span></span>
<span><span class="co">#&gt;  [85] "METAB_VENACV_DA_METAREG"                 </span></span>
<span><span class="co">#&gt;  [86] "METAB_WATSC_DA"                          </span></span>
<span><span class="co">#&gt;  [87] "METAB_WATSC_DA_METAREG"                  </span></span>
<span><span class="co">#&gt;  [88] "METHYL_BAT_NORM_DATA_05FDR"              </span></span>
<span><span class="co">#&gt;  [89] "METHYL_HEART_NORM_DATA_05FDR"            </span></span>
<span><span class="co">#&gt;  [90] "METHYL_HIPPOC_NORM_DATA_05FDR"           </span></span>
<span><span class="co">#&gt;  [91] "METHYL_KIDNEY_NORM_DATA_05FDR"           </span></span>
<span><span class="co">#&gt;  [92] "METHYL_LIVER_NORM_DATA_05FDR"            </span></span>
<span><span class="co">#&gt;  [93] "METHYL_LUNG_NORM_DATA_05FDR"             </span></span>
<span><span class="co">#&gt;  [94] "METHYL_META"                             </span></span>
<span><span class="co">#&gt;  [95] "METHYL_SKMGN_NORM_DATA_05FDR"            </span></span>
<span><span class="co">#&gt;  [96] "METHYL_WATSC_NORM_DATA_05FDR"            </span></span>
<span><span class="co">#&gt;  [97] "OUTLIERS"                                </span></span>
<span><span class="co">#&gt;  [98] "PATHWAY_PARENTS"                         </span></span>
<span><span class="co">#&gt;  [99] "PHENO"                                   </span></span>
<span><span class="co">#&gt; [100] "PHOSPHO_CORTEX_DA"                       </span></span>
<span><span class="co">#&gt; [101] "PHOSPHO_CORTEX_NORM_DATA"                </span></span>
<span><span class="co">#&gt; [102] "PHOSPHO_HEART_DA"                        </span></span>
<span><span class="co">#&gt; [103] "PHOSPHO_HEART_NORM_DATA"                 </span></span>
<span><span class="co">#&gt; [104] "PHOSPHO_KIDNEY_DA"                       </span></span>
<span><span class="co">#&gt; [105] "PHOSPHO_KIDNEY_NORM_DATA"                </span></span>
<span><span class="co">#&gt; [106] "PHOSPHO_LIVER_DA"                        </span></span>
<span><span class="co">#&gt; [107] "PHOSPHO_LIVER_NORM_DATA"                 </span></span>
<span><span class="co">#&gt; [108] "PHOSPHO_LUNG_DA"                         </span></span>
<span><span class="co">#&gt; [109] "PHOSPHO_LUNG_NORM_DATA"                  </span></span>
<span><span class="co">#&gt; [110] "PHOSPHO_META"                            </span></span>
<span><span class="co">#&gt; [111] "PHOSPHO_SKMGN_DA"                        </span></span>
<span><span class="co">#&gt; [112] "PHOSPHO_SKMGN_NORM_DATA"                 </span></span>
<span><span class="co">#&gt; [113] "PHOSPHO_WATSC_DA"                        </span></span>
<span><span class="co">#&gt; [114] "PHOSPHO_WATSC_NORM_DATA"                 </span></span>
<span><span class="co">#&gt; [115] "PROT_CORTEX_DA"                          </span></span>
<span><span class="co">#&gt; [116] "PROT_CORTEX_NORM_DATA"                   </span></span>
<span><span class="co">#&gt; [117] "PROT_HEART_DA"                           </span></span>
<span><span class="co">#&gt; [118] "PROT_HEART_NORM_DATA"                    </span></span>
<span><span class="co">#&gt; [119] "PROT_KIDNEY_DA"                          </span></span>
<span><span class="co">#&gt; [120] "PROT_KIDNEY_NORM_DATA"                   </span></span>
<span><span class="co">#&gt; [121] "PROT_LIVER_DA"                           </span></span>
<span><span class="co">#&gt; [122] "PROT_LIVER_NORM_DATA"                    </span></span>
<span><span class="co">#&gt; [123] "PROT_LUNG_DA"                            </span></span>
<span><span class="co">#&gt; [124] "PROT_LUNG_NORM_DATA"                     </span></span>
<span><span class="co">#&gt; [125] "PROT_META"                               </span></span>
<span><span class="co">#&gt; [126] "PROT_SKMGN_DA"                           </span></span>
<span><span class="co">#&gt; [127] "PROT_SKMGN_NORM_DATA"                    </span></span>
<span><span class="co">#&gt; [128] "PROT_WATSC_DA"                           </span></span>
<span><span class="co">#&gt; [129] "PROT_WATSC_NORM_DATA"                    </span></span>
<span><span class="co">#&gt; [130] "RAT_TO_HUMAN_GENE"                       </span></span>
<span><span class="co">#&gt; [131] "RAT_TO_HUMAN_PHOSPHO"                    </span></span>
<span><span class="co">#&gt; [132] "REPEATED_FEATURES"                       </span></span>
<span><span class="co">#&gt; [133] "REPFDR_INPUTS"                           </span></span>
<span><span class="co">#&gt; [134] "REPFDR_RES"                              </span></span>
<span><span class="co">#&gt; [135] "SEX_COLORS"                              </span></span>
<span><span class="co">#&gt; [136] "TISSUE_ABBREV"                           </span></span>
<span><span class="co">#&gt; [137] "TISSUE_ABBREV_TO_CODE"                   </span></span>
<span><span class="co">#&gt; [138] "TISSUE_CODE_TO_ABBREV"                   </span></span>
<span><span class="co">#&gt; [139] "TISSUE_COLORS"                           </span></span>
<span><span class="co">#&gt; [140] "TISSUE_ORDER"                            </span></span>
<span><span class="co">#&gt; [141] "TRAINING_REGULATED_FEATURES"             </span></span>
<span><span class="co">#&gt; [142] "TRAINING_REGULATED_NORM_DATA"            </span></span>
<span><span class="co">#&gt; [143] "TRAINING_REGULATED_NORM_DATA_NO_OUTLIERS"</span></span>
<span><span class="co">#&gt; [144] "TRNSCRPT_ADRNL_DA"                       </span></span>
<span><span class="co">#&gt; [145] "TRNSCRPT_ADRNL_NORM_DATA"                </span></span>
<span><span class="co">#&gt; [146] "TRNSCRPT_ADRNL_RAW_COUNTS"               </span></span>
<span><span class="co">#&gt; [147] "TRNSCRPT_BAT_DA"                         </span></span>
<span><span class="co">#&gt; [148] "TRNSCRPT_BAT_NORM_DATA"                  </span></span>
<span><span class="co">#&gt; [149] "TRNSCRPT_BAT_RAW_COUNTS"                 </span></span>
<span><span class="co">#&gt; [150] "TRNSCRPT_BLOOD_DA"                       </span></span>
<span><span class="co">#&gt; [151] "TRNSCRPT_BLOOD_NORM_DATA"                </span></span>
<span><span class="co">#&gt; [152] "TRNSCRPT_BLOOD_RAW_COUNTS"               </span></span>
<span><span class="co">#&gt; [153] "TRNSCRPT_COLON_DA"                       </span></span>
<span><span class="co">#&gt; [154] "TRNSCRPT_COLON_NORM_DATA"                </span></span>
<span><span class="co">#&gt; [155] "TRNSCRPT_COLON_RAW_COUNTS"               </span></span>
<span><span class="co">#&gt; [156] "TRNSCRPT_CORTEX_DA"                      </span></span>
<span><span class="co">#&gt; [157] "TRNSCRPT_CORTEX_NORM_DATA"               </span></span>
<span><span class="co">#&gt; [158] "TRNSCRPT_CORTEX_RAW_COUNTS"              </span></span>
<span><span class="co">#&gt; [159] "TRNSCRPT_HEART_DA"                       </span></span>
<span><span class="co">#&gt; [160] "TRNSCRPT_HEART_NORM_DATA"                </span></span>
<span><span class="co">#&gt; [161] "TRNSCRPT_HEART_RAW_COUNTS"               </span></span>
<span><span class="co">#&gt; [162] "TRNSCRPT_HIPPOC_DA"                      </span></span>
<span><span class="co">#&gt; [163] "TRNSCRPT_HIPPOC_NORM_DATA"               </span></span>
<span><span class="co">#&gt; [164] "TRNSCRPT_HIPPOC_RAW_COUNTS"              </span></span>
<span><span class="co">#&gt; [165] "TRNSCRPT_HYPOTH_DA"                      </span></span>
<span><span class="co">#&gt; [166] "TRNSCRPT_HYPOTH_NORM_DATA"               </span></span>
<span><span class="co">#&gt; [167] "TRNSCRPT_HYPOTH_RAW_COUNTS"              </span></span>
<span><span class="co">#&gt; [168] "TRNSCRPT_KIDNEY_DA"                      </span></span>
<span><span class="co">#&gt; [169] "TRNSCRPT_KIDNEY_NORM_DATA"               </span></span>
<span><span class="co">#&gt; [170] "TRNSCRPT_KIDNEY_RAW_COUNTS"              </span></span>
<span><span class="co">#&gt; [171] "TRNSCRPT_LIVER_DA"                       </span></span>
<span><span class="co">#&gt; [172] "TRNSCRPT_LIVER_NORM_DATA"                </span></span>
<span><span class="co">#&gt; [173] "TRNSCRPT_LIVER_RAW_COUNTS"               </span></span>
<span><span class="co">#&gt; [174] "TRNSCRPT_LUNG_DA"                        </span></span>
<span><span class="co">#&gt; [175] "TRNSCRPT_LUNG_NORM_DATA"                 </span></span>
<span><span class="co">#&gt; [176] "TRNSCRPT_LUNG_RAW_COUNTS"                </span></span>
<span><span class="co">#&gt; [177] "TRNSCRPT_META"                           </span></span>
<span><span class="co">#&gt; [178] "TRNSCRPT_OVARY_DA"                       </span></span>
<span><span class="co">#&gt; [179] "TRNSCRPT_OVARY_NORM_DATA"                </span></span>
<span><span class="co">#&gt; [180] "TRNSCRPT_OVARY_RAW_COUNTS"               </span></span>
<span><span class="co">#&gt; [181] "TRNSCRPT_SKMGN_DA"                       </span></span>
<span><span class="co">#&gt; [182] "TRNSCRPT_SKMGN_NORM_DATA"                </span></span>
<span><span class="co">#&gt; [183] "TRNSCRPT_SKMGN_RAW_COUNTS"               </span></span>
<span><span class="co">#&gt; [184] "TRNSCRPT_SKMVL_DA"                       </span></span>
<span><span class="co">#&gt; [185] "TRNSCRPT_SKMVL_NORM_DATA"                </span></span>
<span><span class="co">#&gt; [186] "TRNSCRPT_SKMVL_RAW_COUNTS"               </span></span>
<span><span class="co">#&gt; [187] "TRNSCRPT_SMLINT_DA"                      </span></span>
<span><span class="co">#&gt; [188] "TRNSCRPT_SMLINT_NORM_DATA"               </span></span>
<span><span class="co">#&gt; [189] "TRNSCRPT_SMLINT_RAW_COUNTS"              </span></span>
<span><span class="co">#&gt; [190] "TRNSCRPT_SPLEEN_DA"                      </span></span>
<span><span class="co">#&gt; [191] "TRNSCRPT_SPLEEN_NORM_DATA"               </span></span>
<span><span class="co">#&gt; [192] "TRNSCRPT_SPLEEN_RAW_COUNTS"              </span></span>
<span><span class="co">#&gt; [193] "TRNSCRPT_TESTES_DA"                      </span></span>
<span><span class="co">#&gt; [194] "TRNSCRPT_TESTES_NORM_DATA"               </span></span>
<span><span class="co">#&gt; [195] "TRNSCRPT_TESTES_RAW_COUNTS"              </span></span>
<span><span class="co">#&gt; [196] "TRNSCRPT_VENACV_DA"                      </span></span>
<span><span class="co">#&gt; [197] "TRNSCRPT_VENACV_NORM_DATA"               </span></span>
<span><span class="co">#&gt; [198] "TRNSCRPT_VENACV_RAW_COUNTS"              </span></span>
<span><span class="co">#&gt; [199] "TRNSCRPT_WATSC_DA"                       </span></span>
<span><span class="co">#&gt; [200] "TRNSCRPT_WATSC_NORM_DATA"                </span></span>
<span><span class="co">#&gt; [201] "TRNSCRPT_WATSC_RAW_COUNTS"               </span></span>
<span><span class="co">#&gt; [202] "UBIQ_HEART_DA"                           </span></span>
<span><span class="co">#&gt; [203] "UBIQ_HEART_NORM_DATA"                    </span></span>
<span><span class="co">#&gt; [204] "UBIQ_LIVER_DA"                           </span></span>
<span><span class="co">#&gt; [205] "UBIQ_LIVER_NORM_DATA"                    </span></span>
<span><span class="co">#&gt; [206] "UBIQ_META"</span></span></code></pre></div>
<p>If the <a href="https://motrpac.github.io/MotrpacRatTraining6moData/" class="external-link">MotrpacRatTraining6moData</a>
library is attached, you can learn more about any of these data objects
with <code>?</code>, e.g., <a href="https://motrpac.github.io/MotrpacRatTraining6moData/reference/TISSUE_COLORS.html" class="external-link"><code>?TISSUE_COLORS</code></a>,
and you can load data objects into your environment using
<code><a href="https://rdrr.io/r/utils/data.html" class="external-link">data()</a></code>, e.g.,
<code>data(TRAINING_REGULATED_FEATURES)</code>.</p>
</div>
</div>
<div class="section level2">
<h2 id="analysis">ANALYSIS<a class="anchor" aria-label="anchor" href="#analysis"></a>
</h2>
<div class="section level3">
<h3 id="principal-component-analysis">Principal component analysis<a class="anchor" aria-label="anchor" href="#principal-component-analysis"></a>
</h3>
<p>Function to perform a PCA on any of the available NORM datasets</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Function to perform PCA and plot the first two PCs</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">impute</span><span class="op">)</span>  <span class="co"># For KNN imputation</span></span>
<span></span>
<span><span class="va">do_pca_plot</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">norm_data_object</span>, <span class="va">object_name</span> <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">impute</span> <span class="op">=</span> <span class="cn">FALSE</span>, <span class="va">remove_outliers</span> <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Check if the object name is provided or extract it</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">object_name</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">object_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/deparse.html" class="external-link">deparse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/substitute.html" class="external-link">substitute</a></span><span class="op">(</span><span class="va">norm_data_object</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  </span>
<span>  <span class="co"># Check if the object name contains 'NORM'</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"NORM"</span>, <span class="va">object_name</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"The provided object does not contain 'NORM' in its name. Please provide a valid NORM_DATA object."</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  </span>
<span>  <span class="co"># Extract the sample IDs from the list</span></span>
<span>  <span class="va">sample_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">norm_data_object</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"feature"</span>, <span class="st">"feature_ID"</span>, <span class="st">"tissue"</span>, <span class="st">"assay"</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Subset the data using sample_ids</span></span>
<span>  <span class="va">data_subset</span> <span class="op">&lt;-</span> <span class="va">norm_data_object</span><span class="op">[</span>, <span class="va">sample_ids</span><span class="op">]</span></span>
<span>  </span>
<span>  <span class="co"># Remove columns with more than 80% missing values</span></span>
<span>  <span class="va">valid_cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colMeans</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">data_subset</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">0.8</span></span>
<span>  <span class="va">data_subset</span> <span class="op">&lt;-</span> <span class="va">data_subset</span><span class="op">[</span>, <span class="va">valid_cols</span><span class="op">]</span></span>
<span>  </span>
<span>  <span class="co"># Ensure there are enough columns left for PCA</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">data_subset</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">2</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"Not enough columns remaining after filtering for PCA."</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  </span>
<span>  <span class="co"># Impute or clean data based on 'impute' argument</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="va">impute</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="co"># Impute missing values using KNN, suppressing messages</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/sink.html" class="external-link">sink</a></span><span class="op">(</span>file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/showConnections.html" class="external-link">nullfile</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="va">data_imputed</span> <span class="op">&lt;-</span> <span class="fu">impute</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/impute/man/impute.knn.html" class="external-link">impute.knn</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">data_subset</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">data</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/sink.html" class="external-link">sink</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>    <span class="co"># Remove rows with NA or Inf values</span></span>
<span>    <span class="va">data_imputed</span> <span class="op">&lt;-</span> <span class="va">data_subset</span><span class="op">[</span><span class="fu">stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/stats/complete.cases.html" class="external-link">complete.cases</a></span><span class="op">(</span><span class="va">data_subset</span><span class="op">)</span> <span class="op">&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">data_subset</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/is.finite.html" class="external-link">is.finite</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, <span class="op">]</span></span>
<span>  <span class="op">}</span></span>
<span>  </span>
<span>  <span class="co"># Ensure there are enough rows left for PCA</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">data_imputed</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">2</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"Not enough rows remaining after data cleaning for PCA."</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  </span>
<span>  <span class="co"># Update the sample_ids based on the final data used in PCA</span></span>
<span>  <span class="va">valid_sample_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">data_imputed</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Check if sample_ids match 'pid' or 'viallabel' in PHENO and use them accordingly</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="va">valid_sample_ids</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">PHENO</span><span class="op">$</span><span class="va">pid</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">df_sample_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">valid_sample_ids</span>, <span class="va">PHENO</span><span class="op">$</span><span class="va">pid</span><span class="op">)</span></span>
<span>  <span class="op">}</span> <span class="kw">else</span> <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="va">valid_sample_ids</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">PHENO</span><span class="op">$</span><span class="va">viallabel</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">df_sample_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">valid_sample_ids</span>, <span class="va">PHENO</span><span class="op">$</span><span class="va">viallabel</span><span class="op">)</span></span>
<span>  <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"Sample IDs do not match either 'pid' or 'viallabel' in PHENO."</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  </span>
<span>  <span class="co"># Perform PCA on the transposed imputed/cleaned data</span></span>
<span>  <span class="va">pca_result</span> <span class="op">&lt;-</span> <span class="fu">stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/stats/prcomp.html" class="external-link">prcomp</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">data_imputed</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Calculate the percentage of variance explained by each PC</span></span>
<span>  <span class="va">variance_explained</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">pca_result</span><span class="op">)</span><span class="op">$</span><span class="va">importance</span><span class="op">[</span><span class="fl">2</span>, <span class="op">]</span> <span class="op">*</span> <span class="fl">100</span></span>
<span>  </span>
<span>  <span class="co"># Create a data frame with phenotypic data and the first 3 PCs</span></span>
<span>  <span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    group <span class="op">=</span> <span class="va">PHENO</span><span class="op">[</span><span class="va">df_sample_ids</span>, <span class="st">"group"</span><span class="op">]</span>,</span>
<span>    sex <span class="op">=</span> <span class="va">PHENO</span><span class="op">[</span><span class="va">df_sample_ids</span>, <span class="st">"sex"</span><span class="op">]</span>,</span>
<span>    <span class="va">pca_result</span><span class="op">$</span><span class="va">x</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span> <span class="co"># take the first three principal components</span></span>
<span>  <span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Remove extreme outliers based on IQR for PC1 and PC2 if 'remove_outliers' is TRUE</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="va">remove_outliers</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw">for</span> <span class="op">(</span><span class="va">pc</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PC1"</span>, <span class="st">"PC2"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="va">Q1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">df</span><span class="op">[[</span><span class="va">pc</span><span class="op">]</span><span class="op">]</span>, <span class="fl">0.25</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>      <span class="va">Q3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">df</span><span class="op">[[</span><span class="va">pc</span><span class="op">]</span><span class="op">]</span>, <span class="fl">0.75</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>      <span class="va">IQR_value</span> <span class="op">&lt;-</span> <span class="va">Q3</span> <span class="op">-</span> <span class="va">Q1</span></span>
<span>      </span>
<span>      <span class="co"># Define the boundaries</span></span>
<span>      <span class="va">lower_bound</span> <span class="op">&lt;-</span> <span class="va">Q1</span> <span class="op">-</span> <span class="fl">3</span> <span class="op">*</span> <span class="va">IQR_value</span></span>
<span>      <span class="va">upper_bound</span> <span class="op">&lt;-</span> <span class="va">Q3</span> <span class="op">+</span> <span class="fl">3</span> <span class="op">*</span> <span class="va">IQR_value</span></span>
<span>      </span>
<span>      <span class="co"># Filter the data to remove outliers</span></span>
<span>      <span class="va">df</span> <span class="op">&lt;-</span> <span class="va">df</span><span class="op">[</span><span class="va">df</span><span class="op">[[</span><span class="va">pc</span><span class="op">]</span><span class="op">]</span> <span class="op">&gt;=</span> <span class="va">lower_bound</span> <span class="op">&amp;</span> <span class="va">df</span><span class="op">[[</span><span class="va">pc</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;=</span> <span class="va">upper_bound</span>, <span class="op">]</span></span>
<span>    <span class="op">}</span></span>
<span>  <span class="op">}</span></span>
<span>  </span>
<span>  <span class="co"># Ensure that there are still data points left to plot</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">1</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"No data points left to plot after removing outliers."</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  </span>
<span>  <span class="co"># Plot the first two PCs with variance explained in axis labels</span></span>
<span>  <span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">PC1</span>, y <span class="op">=</span> <span class="va">PC2</span>, fill <span class="op">=</span> <span class="va">group</span>, shape <span class="op">=</span> <span class="va">sex</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">3</span>, colour <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">GROUP_COLORS</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_shape_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>male <span class="op">=</span> <span class="fl">21</span>, female <span class="op">=</span> <span class="fl">24</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span>override.aes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>shape <span class="op">=</span> <span class="fl">21</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">object_name</span><span class="op">)</span>,</span>
<span>         x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"PC1 ("</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">variance_explained</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="fl">2</span><span class="op">)</span>, <span class="st">"% Variance)"</span><span class="op">)</span>,</span>
<span>         y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"PC2 ("</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">variance_explained</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, <span class="fl">2</span><span class="op">)</span>, <span class="st">"% Variance)"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="section level4">
<h4 id="one-tissue-at-a-time">One tissue at a time<a class="anchor" aria-label="anchor" href="#one-tissue-at-a-time"></a>
</h4>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example usage of the function</span></span>
<span><span class="fu">do_pca_plot</span><span class="op">(</span><span class="va">ATAC_LIVER_NORM_DATA_05FDR</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-16-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">do_pca_plot</span><span class="op">(</span><span class="va">ATAC_LIVER_NORM_DATA_05FDR</span>, remove_outliers <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-16-2.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu">do_pca_plot</span><span class="op">(</span><span class="va">TRNSCRPT_LIVER_NORM_DATA</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-16-3.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">do_pca_plot</span><span class="op">(</span><span class="va">TRNSCRPT_LIVER_NORM_DATA</span>, remove_outliers <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-16-4.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu">do_pca_plot</span><span class="op">(</span><span class="va">PROT_LIVER_NORM_DATA</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-16-5.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">do_pca_plot</span><span class="op">(</span><span class="va">PHOSPHO_LIVER_NORM_DATA</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-16-6.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="all-tissues-assays-except-metabolomics">All tissues / Assays (except Metabolomics)<a class="anchor" aria-label="anchor" href="#all-tissues-assays-except-metabolomics"></a>
</h4>
<p>Perform PCA and plot for all objects in the list of norm objects
generated above</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span> <span class="op">(</span><span class="va">object_name</span> <span class="kw">in</span> <span class="va">norm_objects</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">norm_data_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html" class="external-link">get</a></span><span class="op">(</span><span class="va">object_name</span><span class="op">)</span></span>
<span>  <span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">do_pca_plot</span><span class="op">(</span><span class="va">norm_data_object</span>, <span class="va">object_name</span>, impute <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-1.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-2.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-3.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-4.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-5.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-6.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-7.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-8.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-9.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-10.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-11.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-12.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-13.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-14.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-15.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-16.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-17.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-18.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-19.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-20.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-21.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-22.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-23.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-24.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-25.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-26.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-27.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-28.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-29.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-30.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-31.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-32.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-33.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-34.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-35.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-36.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-37.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-38.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-39.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-40.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-41.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-42.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-43.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-44.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-45.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-46.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-47.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-48.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-49.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-50.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-51.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-52.png" width="700" style="display: block; margin: auto;"><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-17-53.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="metabolomics">Metabolomics<a class="anchor" aria-label="anchor" href="#metabolomics"></a>
</h4>
<p>The <code>NORM</code> data object for metabolomics contains all
tissues and assays. To perform a PCA, it requires previous filtering and
assay selection. Let’s massage the data to get a PCA for every assay
available for <code>LIVER</code>. Some of the metabolomics assays
require imputation due to a larger number of missing values.</p>
</div>
</div>
<div class="section level3">
<h3 id="differential-analysis">Differential analysis<a class="anchor" aria-label="anchor" href="#differential-analysis"></a>
</h3>
<p>More details about the differential analysis methods are available in
the <a href="https://www.nature.com/articles/s41586-023-06877-w#Sec14" class="external-link">supplementary
information of our <em>Nature</em> publication</a>.</p>
<p>Simply put:</p>
<ul>
<li>the <em>training</em> differential analysis considers all training
groups for each sex (sedentary controls and 4 training time points) to
determine if the analyte significantly changes in either sex at any
point during the training time course. The adjusted p-values from this
analysis were used to determine the set of analytes that are regulated
by endurance exercise training at 5% FDR, referred to as the
<em>training-regulated features</em>.</li>
<li>The <em>timewise</em> differential analysis performs pairwise
contrasts between trained animals at each time point (1, 2, 4, or 8
weeks) and the sex-matched sedentary control animals. This gives us sex-
and time- specific p-values and effect sizes, referred to as the
<em>timewise summary statistics</em>.</li>
</ul>
<div class="section level4">
<h4 id="volcano-plots">Volcano plots<a class="anchor" aria-label="anchor" href="#volcano-plots"></a>
</h4>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Function to generate a volcano plot for a given data object</span></span>
<span><span class="va">create_volcano_plot</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">data_object</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Extract the name of the data object</span></span>
<span>  <span class="va">object_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/deparse.html" class="external-link">deparse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/substitute.html" class="external-link">substitute</a></span><span class="op">(</span><span class="va">data_object</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Filter out outliers with absolute logFC greater than 10</span></span>
<span>  <span class="va">filtered_data</span> <span class="op">&lt;-</span> <span class="va">data_object</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">logFC</span><span class="op">)</span> <span class="op">&lt;=</span> <span class="fl">10</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">filtered_data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">logFC</span>, y <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">adj_p_value</span><span class="op">)</span>, color <span class="op">=</span> <span class="va">adj_p_value</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.6</span>, size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TRUE"</span> <span class="op">=</span> <span class="st">"red"</span>, <span class="st">"FALSE"</span> <span class="op">=</span> <span class="st">"grey"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span><span class="va">comparison_group</span><span class="op">~</span><span class="va">sex</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_linedraw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>      title <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">object_name</span><span class="op">)</span>,</span>
<span>      x <span class="op">=</span> <span class="st">"logFC"</span>,</span>
<span>      y <span class="op">=</span> <span class="st">"-log10(Adjusted P-value)"</span>,</span>
<span>      color <span class="op">=</span> <span class="st">"Significant (FDR &lt; 0.05)"</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>      plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">16</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,</span>
<span>      axis.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>,</span>
<span>      axis.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>      strip.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>,</span>
<span>      legend.position <span class="op">=</span> <span class="st">"bottom"</span></span>
<span>    <span class="op">)</span></span>
<span>  </span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Example usage</span></span>
<span><span class="fu">create_volcano_plot</span><span class="op">(</span><span class="va">TRNSCRPT_BAT_DA</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-19-1.png" width="768" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="replicating-differential-analysis-results">Replicating differential analysis results<a class="anchor" aria-label="anchor" href="#replicating-differential-analysis-results"></a>
</h4>
<p>We provide the following functions to replicate our <em>timewise</em>
and <em>training</em> differential analysis results for each dataset.
Look at the corresponding documentation for each function for more
details, e.g. <code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/transcript_timewise_da.html" class="external-link">?transcript_timewise_da</a></code>.</p>
<ul>
<li>Proteomics (ACETYL, PHOSPHO, PROT, UBIQ)
<ul>
<li>
<code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/proteomics_timewise_da.html" class="external-link">proteomics_timewise_da()</a></code><br>
</li>
<li>
<code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/proteomics_training_da.html" class="external-link">proteomics_training_da()</a></code><br>
</li>
</ul>
</li>
<li>ATAC
<ul>
<li>
<code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/atac_training_da.html" class="external-link">atac_training_da()</a></code><br>
</li>
<li>
<code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/atac_timewise_da.html" class="external-link">atac_timewise_da()</a></code><br>
</li>
</ul>
</li>
<li>IMMUNO
<ul>
<li>
<code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/immuno_timewise_da.html" class="external-link">immuno_timewise_da()</a></code><br>
</li>
<li>
<code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/immuno_training_da.html" class="external-link">immuno_training_da()</a></code><br>
</li>
</ul>
</li>
<li>METAB
<ul>
<li>
<code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/metab_timewise_da.html" class="external-link">metab_timewise_da()</a></code><br>
</li>
<li>
<code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/metab_training_da.html" class="external-link">metab_training_da()</a></code><br>
</li>
<li>
<code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/metab_meta_regression.html" class="external-link">metab_meta_regression()</a></code><br>
</li>
</ul>
</li>
<li>METHYL
<ul>
<li>
<code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/rrbs_differential_analysis.html" class="external-link">rrbs_differential_analysis()</a></code><br>
</li>
</ul>
</li>
<li>TRNSCRPT
<ul>
<li>
<code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/transcript_timewise_da.html" class="external-link">transcript_timewise_da()</a></code><br>
</li>
<li><code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/transcript_training_da.html" class="external-link">transcript_training_da()</a></code></li>
</ul>
</li>
</ul>
<p>Here, we replicate the protein acetylation timewise differential
analysis results provided in <a href="https://motrpac.github.io/MotrpacRatTraining6moData/" class="external-link">MotrpacRatTraining6moData</a>.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">timewiseList</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">tissue</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HEART"</span>,<span class="st">"LIVER"</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">timewiseList</span><span class="op">[[</span><span class="va">tissue</span><span class="op">]</span><span class="op">]</span> <span class="op">=</span> <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/proteomics_timewise_da.html" class="external-link">proteomics_timewise_da</a></span><span class="op">(</span><span class="st">"ACETYL"</span>, <span class="va">tissue</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="co">#&gt; ACETYL_HEART_NORM_DATA</span></span>
<span><span class="co">#&gt; Warning: Partial NA coefficients for 157 probe(s)</span></span>
<span><span class="co">#&gt; ACETYL_LIVER_NORM_DATA</span></span>
<span><span class="co">#&gt; Warning: Partial NA coefficients for 355 probe(s)</span></span>
<span><span class="va">timewise</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="st">"rbind"</span>, <span class="va">timewiseList</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># merge with version of results in MotrpacRatTraining6moData</span></span>
<span><span class="va">original</span> <span class="op">=</span> <span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/combine_da_results.html" class="external-link">combine_da_results</a></span><span class="op">(</span>assays<span class="op">=</span><span class="st">"ACETYL"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ACETYL_HEART_DA</span></span>
<span><span class="co">#&gt; ACETYL_LIVER_DA</span></span>
<span><span class="va">merged</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">original</span>, <span class="va">timewise</span>, </span>
<span>               by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"feature_ID"</span>,<span class="st">"assay"</span>,<span class="st">"assay_code"</span>,<span class="st">"tissue"</span>,<span class="st">"tissue_code"</span>,<span class="st">"sex"</span>,<span class="st">"comparison_group"</span><span class="op">)</span>,</span>
<span>               suffixes<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"_orginal"</span>,<span class="st">"_reproduced"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">merged</span><span class="op">$</span><span class="va">p_value_orginal</span><span class="op">)</span>, </span>
<span>     <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">merged</span><span class="op">$</span><span class="va">p_value_reproduced</span><span class="op">)</span>, </span>
<span>     type<span class="op">=</span><span class="st">"p"</span>, cex<span class="op">=</span><span class="fl">0.5</span>,</span>
<span>     xlab<span class="op">=</span><span class="st">"Original timewise p-value (-log10)"</span>,</span>
<span>     ylab<span class="op">=</span><span class="st">"Reproduced timewise p-value (-log10)"</span>,</span>
<span>     main<span class="op">=</span><span class="st">"Timewise DA results for ACETYL data"</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/replicate%20prot-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="plot-da-for-a-single-feature">Plot DA for a single feature<a class="anchor" aria-label="anchor" href="#plot-da-for-a-single-feature"></a>
</h4>
<p>We can plot the differential analysis results for a single feature
using <code><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/plot_feature_logfc.html" class="external-link">plot_feature_logfc()</a></code>. All of the following examples
are different ways to plot results for the same feature.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/plot_feature_logfc.html" class="external-link">plot_feature_logfc</a></span><span class="op">(</span>feature <span class="op">=</span> <span class="st">"PROT;SKM-GN;NP_786937.1"</span>,</span>
<span>                   add_gene_symbol <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-20-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/plot_feature_logfc.html" class="external-link">plot_feature_logfc</a></span><span class="op">(</span>assay <span class="op">=</span> <span class="st">"PROT"</span>,</span>
<span>                   tissue <span class="op">=</span> <span class="st">"SKM-GN"</span>,</span>
<span>                   feature_ID <span class="op">=</span> <span class="st">"NP_786937.1"</span>,</span>
<span>                   scale_x_by_time <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-20-2.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://motrpac.github.io/MotrpacRatTraining6mo/reference/plot_feature_logfc.html" class="external-link">plot_feature_logfc</a></span><span class="op">(</span>assay <span class="op">=</span> <span class="st">"PROT"</span>,</span>
<span>                   tissue <span class="op">=</span> <span class="st">"SKM-GN"</span>,</span>
<span>                   feature_ID <span class="op">=</span> <span class="st">"NP_786937.1"</span>,</span>
<span>                   facet_by_sex <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                   add_adj_p <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="rat-endurance-6m_files/figure-html/unnamed-chunk-20-3.png" width="700" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level3">
<h3 id="bayesian-graphical-clustering">Bayesian graphical clustering<a class="anchor" aria-label="anchor" href="#bayesian-graphical-clustering"></a>
</h3>
<p>For a detail tutorial on the graphical clustering analysis, please,
visit the original vignette part of the MotrpacRatTraining6mo package <a href="https://motrpac.github.io/MotrpacRatTraining6mo/articles/MotrpacRatTraining6mo.html#bayesian-graphical-clustering" class="external-link">here</a>.</p>
</div>
<div class="section level3">
<h3 id="getting-help">Getting help<a class="anchor" aria-label="anchor" href="#getting-help"></a>
</h3>
<p>For questions, bug reporting, and data requests for the
<code>MotrpacRatTraining6moData</code> package, please <a href="https://github.com/MoTrPAC/MotrpacRatTraining6moData/issues" class="external-link">submit
a new issue here</a> and include as many details as possible.</p>
<p>If the issue is related to functions provided in the
<code>MotrpacRatTraining6mo</code> package, please submit an issue <a href="https://github.com/MoTrPAC/MotrpacRatTraining6mo/issues" class="external-link">here</a>.</p>
</div>
<div class="section level3">
<h3 id="acknowledgements">Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"></a>
</h3>
<p>MoTrPAC is supported by the National Institutes of Health (NIH)
Common Fund through cooperative agreements managed by the National
Institute of Diabetes and Digestive and Kidney Diseases (NIDDK),
National Institute of Arthritis and Musculoskeletal Diseases (NIAMS),
and National Institute on Aging (NIA).</p>
</div>
<div class="section level3">
<h3 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h3>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.3 (2024-02-29)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: macOS 15.4.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/Los_Angeles</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] UpSetR_1.4.0                    viridis_0.6.5                  </span></span>
<span><span class="co">#&gt;  [3] viridisLite_0.4.2               pheatmap_1.0.12                </span></span>
<span><span class="co">#&gt;  [5] lubridate_1.9.4                 forcats_1.0.0                  </span></span>
<span><span class="co">#&gt;  [7] stringr_1.5.1                   purrr_1.0.4                    </span></span>
<span><span class="co">#&gt;  [9] readr_2.1.5                     tidyr_1.3.1                    </span></span>
<span><span class="co">#&gt; [11] tibble_3.2.1                    tidyverse_2.0.0                </span></span>
<span><span class="co">#&gt; [13] impute_1.76.0                   dplyr_1.1.4                    </span></span>
<span><span class="co">#&gt; [15] ggplot2_3.5.1                   MotrpacRatTraining6mo_1.6.6    </span></span>
<span><span class="co">#&gt; [17] MotrpacRatTraining6moData_2.0.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] Rdpack_2.6.2        mnormt_2.1.1        gridExtra_2.3      </span></span>
<span><span class="co">#&gt;  [4] sandwich_3.1-1      rlang_1.1.5         magrittr_2.0.3     </span></span>
<span><span class="co">#&gt;  [7] multcomp_1.4-28     qqconf_1.3.2        compiler_4.3.3     </span></span>
<span><span class="co">#&gt; [10] systemfonts_1.1.0   vctrs_0.6.5         crayon_1.5.3       </span></span>
<span><span class="co">#&gt; [13] pkgconfig_2.0.3     fastmap_1.2.0       labeling_0.4.3     </span></span>
<span><span class="co">#&gt; [16] ggraph_2.2.1        r2r_0.1.2           rmarkdown_2.29     </span></span>
<span><span class="co">#&gt; [19] tzdb_0.4.0          ragg_1.3.3          xfun_0.52          </span></span>
<span><span class="co">#&gt; [22] cachem_1.1.0        jsonlite_1.8.9      tweenr_2.0.3       </span></span>
<span><span class="co">#&gt; [25] R6_2.6.1            bslib_0.9.0         stringi_1.8.4      </span></span>
<span><span class="co">#&gt; [28] RColorBrewer_1.1-3  limma_3.58.1        mutoss_0.1-13      </span></span>
<span><span class="co">#&gt; [31] jquerylib_0.1.4     numDeriv_2016.8-1.1 Rcpp_1.0.14        </span></span>
<span><span class="co">#&gt; [34] knitr_1.49          zoo_1.8-12          Matrix_1.6-5       </span></span>
<span><span class="co">#&gt; [37] splines_4.3.3       igraph_2.1.4        timechange_0.3.0   </span></span>
<span><span class="co">#&gt; [40] tidyselect_1.2.1    rstudioapi_0.17.1   yaml_2.3.10        </span></span>
<span><span class="co">#&gt; [43] codetools_0.2-20    plyr_1.8.9          lattice_0.22-6     </span></span>
<span><span class="co">#&gt; [46] Biobase_2.66.0      withr_3.0.2         evaluate_1.0.3     </span></span>
<span><span class="co">#&gt; [49] desc_1.4.3          survival_3.8-3      polyclip_1.10-7    </span></span>
<span><span class="co">#&gt; [52] pillar_1.10.1       stats4_4.3.3        sn_2.1.1           </span></span>
<span><span class="co">#&gt; [55] generics_0.1.3      mathjaxr_1.6-0      hms_1.1.3          </span></span>
<span><span class="co">#&gt; [58] munsell_0.5.1       scales_1.3.0        TFisher_0.2.0      </span></span>
<span><span class="co">#&gt; [61] glue_1.8.0          tools_4.3.3         metap_1.11         </span></span>
<span><span class="co">#&gt; [64] data.table_1.16.4   locfit_1.5-9.11     fs_1.6.5           </span></span>
<span><span class="co">#&gt; [67] visNetwork_2.1.2    mvtnorm_1.3-3       graphlayouts_1.2.2 </span></span>
<span><span class="co">#&gt; [70] tidygraph_1.3.1     grid_4.3.3          plotrix_3.8-4      </span></span>
<span><span class="co">#&gt; [73] rbibutils_2.3       edgeR_4.0.16        colorspace_2.1-1   </span></span>
<span><span class="co">#&gt; [76] ggforce_0.4.2       cli_3.6.4           textshaping_1.0.0  </span></span>
<span><span class="co">#&gt; [79] gtable_0.3.6        sass_0.4.9          digest_0.6.37      </span></span>
<span><span class="co">#&gt; [82] BiocGenerics_0.48.1 ggrepel_0.9.6       TH.data_1.1-3      </span></span>
<span><span class="co">#&gt; [85] htmlwidgets_1.6.4   farver_2.1.2        memoise_2.0.1      </span></span>
<span><span class="co">#&gt; [88] htmltools_0.5.8.1   pkgdown_2.1.1       multtest_2.58.0    </span></span>
<span><span class="co">#&gt; [91] lifecycle_1.0.4     statmod_1.5.0       MASS_7.3-60.0.1</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by David Jimenez-Morales.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
